---
title: "RNAseq Processing and QC"
author: "Annat Haber"
date: '`r Sys.Date()`'
output:
  html_document:
    toc: true
    code_folding: hide
    df_print: "kable"
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = TRUE)
```

```{r libraries}
suppressPackageStartupMessages({
  library(tidyverse)
  library(synapser)
})
```

```{r metadata}
meta <- read_csv("output/RNAseq/RNAseq_sample_meta.csv", col_types = cols(.default = "c")) %>% 
  arrange(genotype,age,sex)
```

## Linking input fastq files
Creating symbolic links to the original fastq files.  
Files are renamed as `<sample_id>_R[1|2].fastq.gz`. Sample ids are padded with zeros to have five digits      
```{r symlink_data, eval=FALSE}
# original files to link to
origfiles <- c(list.files("/projects/howell-lab/00_fastq/reagaa/RNAseq",
                        pattern=".*GT20.*.fastq.gz", full.names = TRUE))

for (id in meta$tag_id) {
  if (id=="48289") { ## typo in the fastq file name
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R1.fastq.gz") # link name and path
      ofile <- grep("/49289_.+_R1", origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
      
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R2.fastq.gz") # link name and path
      ofile <- grep("/49289_.+_R2", origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)

  } else {
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R1.fastq.gz") # link name and path
      ofile <- grep(paste0("/",id,"_.+_R1"), origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
      
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R2.fastq.gz") # link name and path
      ofile <- grep(paste0("/",id,"_.+_R2"), origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
      }
}
```

Creating [`samplesheet.csv`](https://github.com/nf-core/rnaseq/blob/3.0/assets/samplesheet.csv) for the nf-core rnaseq pipeline
```{r, eval=FALSE}
nr <- dplyr::count(meta, group) %>% pull(n)
repl <- c(); for (i in 1:length(nr)) {repl <- c(repl, 1:nr[i])}

fqpath <- "/projects/carter-lab/MTHFR/data/RNAseq/"
  
samplesheet <- tibble(group = meta$group,
                      replicate = repl,
                      fastq_1 = paste0(fqpath, meta$sample_id, "_R1.fastq.gz"),
                      fastq_2 = paste0(fqpath, meta$sample_id, "_R2.fastq.gz"),
                      strandedness = "reverse")

write_csv(samplesheet, "output/RNAseq/1_Processing_QC/samplesheet.csv")

meta$samplesheet <- paste0(meta$group, "_R", repl)
```

## Processing fastq Files
The fastq files are processed using the [nextflow-core methylseq pipeline](https://github.com/nf-core/methylseq), and submitted to sumner with [this script](assets/RRBS_run_nf_methylseq.sbatch).  
Reads were aligned to GRCm38 reference genome as a directional library. 
```{bash run_nf, eval=FALSE}
cd /projects/carter-lab/MTHFR/output/RNAseq/1_Processing_QC
sbatch run_nf_rnaseq.sbatch
cp run_nf_rnaseq.sbatch /projects/carter-lab/MTHFR/assets/RNAseq_run_nf_rnaseq.sbatch
nextflow clean -f
```

## QC: Processing Pipeline
Summary reports generated by the pipeline:
```{r qc_files, results='hide'}
file.copy(from = "output/RNAseq/1_Processing_QC/nf_rnaseq_output/multiqc/star_rsem/multiqc_report.html", 
          to = "./assets/RNAseq_multiqc_report.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)

file.copy(from = "output/RNAseq/1_Processing_QC/samplesheet.csv", 
          to = "./assets/RNAseq_samplesheet.csv", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
```
1. [Overview of pipeline and output](https://nf-co.re/rnaseq/3.0/output)  
2. [QC report for all steps](assets/RNAseq_multiqc_report.html)   
3. [Sample list that matches the pipeline's notation](assets/RNAseq_samplesheet.csv)

## QC: Genotypes
```{bash mpileup, eval=FALSE}
#singularity pull --name ~/bin/bam-readcount shub://jaxreg.jax.org/rit-ci/bam-readcount:v0.8
#~/bin/bam-readcount $inpath/CC_18_F_R1.sorted.bam 4:148048151-148048151

dir=/projects/carter-lab/MTHFR/output/RNAseq/1_Processing_QC
ls $dir/nf_rnaseq_output/star_rsem/*[1-7].sorted.bam > $dir/bamlist.txt

singularity exec ~/bin/htslib samtools mpileup \
      -B \
      -r 4:148048141-148048161 \
      -b $dir/bamlist.txt > $dir/mp_gt.txt
```

```{r parse_mpgt}
mpgt <- read_tsv("output/RNAseq/1_Processing_QC/mp_gt.txt", col_names = FALSE)
mpgt <- select(mpgt, seq(5, ncol(mpgt), 3))

#bamlist <- read_table("output/RNAseq/1_Processing_QC/bamlist.txt", col_names = FALSE)

gt_c667t <- slice(mpgt, 11) %>%
  t() %>%
  as.tibble() %>%
  mutate("A" = str_count(V1, "[a|A]"),
         "C" = str_count(V1, "[c|C]"),
         "G" = str_count(V1, "[g|G]"),
         "T" = str_count(V1, "[t|T]")) %>%
  select(-V1) %>%
  add_column(sample_id = meta$sample_id, group = meta$group, .before = 1)

gt_c667t
```

```{r, include=FALSE, eval=FALSE}
gt7 <- mpgt %>%
  mutate(across(.cols = everything(), .fns = substr, start=5, stop=5)) %>%
  mutate(across(.cols = everything(), .fns = str_to_upper)) %>%
  t() %>%
  as.tibble() %>%
  unite(col=gt_141_161, 1:21, sep="")
```

## Output
Saving
```{r output}
saveRDS(meta, file="output/RNAseq/1_Processing_QC/samplesheet_meta.RDS")
```

Uploading to [Synapse](https://www.synapse.org/#!Synapse:syn23573590/wiki/607402) as csv file.
```{r synapse, results='hide'}
synLogin() 


```
