---
title: "RNAseq Differential Expression"
author: "Annat Haber"
date: "5/1/2021"
output:
  html_document:
    toc: true
    code_folding: hide
    df_print: "kable"
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = TRUE)
```

```{r libraries}
suppressPackageStartupMessages({
  library(tidyverse)
  library(synapser)
  library(limma)
  library(Glimma)
  library(edgeR)
  library(Mus.musculus)  })
```

## Input
Read sample metadata and raw counts.  
Sort everything by sample ID.  
Filter out genes that have less than 100x reads total.
```{r}
# read sample info
meta <- readRDS("output/RNAseq/1_Processing_QC/samplesheet_meta.RDS")

# get mapping between pipeline sample id and original sample id
idkey <- read_csv("output/RNAseq/1_Processing_QC/nf_rnaseq_output/pipeline_info/samplesheet.valid.csv") 
idkey$sample <- sub("_T1", "", idkey$sample)
idkey$sample_id <- sub("(\\d+)_R1.fastq.gz", "\\1", basename(idkey$fastq_1))
idkey <- idkey[,c("sample", "sample_id")] %>%
  arrange(sample_id)

# read raw counts
ctsfile <- "output/RNAseq/1_Processing_QC/nf_rnaseq_output/star_rsem/rsem.merged.gene_counts.tsv"
counts <- read_tsv(ctsfile) %>%
  dplyr::select(-"transcript_id(s)") %>%
  column_to_rownames("gene_id") %>%
  round() %>%
  as.matrix()

# replace pipeline sample id with original sample id
counts <- counts[,idkey$sample]
colnames(counts) <- idkey$sample_id
# Filter out genes that have less than 100x reads total
counts <- counts[rowSums(counts) >100,]
```

## DESeq objects and results {.tabset .tabset-fade .tabset-pills}
For each combination of sex (M,F) and age (6,18) we compare CT to CC and TT to CC.  
Genes with p-adj < 0.1 are in blue.
```{r deseq_obj}
#grp <- unique(meta$group[meta$genotype!="CC"])
grp <- meta %>%
  filter(genotype!="CC") %>%
  distinct(genotype, sex, age)

ddsL <- de_resL <- list()

for (i in 1:nrow(grp)) {
  gt <- grp$genotype[i]
  sx <- grp$sex[i]
  ag <- grp$age[i]
  gr <- paste(gt, sx, ag, sep="_")
  
  coldata <- meta %>%
    filter(genotype %in% c("CC",gt) & sex == sx & age == ag) %>%
    select(genotype, sample_id) %>%
    column_to_rownames("sample_id")

  cts <- counts[, rownames(coldata)]

  dds <- DESeqDataSetFromMatrix(cts, coldata, design = ~ genotype)
  dds$genotype <- relevel(dds$genotype, ref = "CC")
 
  ddsL[[gr]] <- dds <- DESeq(dds)
  res <- results(dds)
  res <- res[order(res),]
  de_resL[[gr]] <- res
  
  rm(gt, sx, ag, coldata, cts, dds, res)
}
```

```{r, deseq_res, results='asis', cache=TRUE}
for(gr in names(de_resL)){
  cat("### ", gr ,"\n")
  res <- de_resL[[gr]]
  #cat(summary(res))
  cat(plotMA(res), "\n")
  cat("\n", "significant genes:\n", rownames(res[which(res$padj<0.1),]))
  cat('\n\n')
}
```

## Output
Uploading to synapse
```{r}

```

