---
title: "RNAseq Processing and QC"
author: "Annat Haber"
date: '2021-05-19'
output:
  html_document:
    toc: true
    code_folding: hide
    df_print: "kable"
---


```r
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = TRUE)
```


```r
suppressPackageStartupMessages({
  library(tidyverse)
  library(synapser)
})
```


```r
meta <- read_csv("output/RNAseq/RNAseq_sample_meta.csv", col_types = cols(.default = "c")) %>% 
  arrange(genotype,age,sex)
```

## Linking input fastq files
Creating symbolic links to the original fastq files.  
Files are renamed as `<sample_id>_R[1|2].fastq.gz`. Sample ids are padded with zeros to have five digits      

```r
# original files to link to
origfiles <- c(list.files("/projects/howell-lab/00_fastq/reagaa/RNAseq",
                        pattern=".*GT20.*.fastq.gz", full.names = TRUE))

for (id in meta$tag_id) {
  if (id=="48289") { ## typo in the fastq file name
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R1.fastq.gz") # link name and path
      ofile <- grep("/49289_.+_R1", origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
      
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R2.fastq.gz") # link name and path
      ofile <- grep("/49289_.+_R2", origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)

  } else {
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R1.fastq.gz") # link name and path
      ofile <- grep(paste0("/",id,"_.+_R1"), origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
      
      flink <- paste0("data/RNAseq/", str_pad(id, 5, pad = "0"), "_R2.fastq.gz") # link name and path
      ofile <- grep(paste0("/",id,"_.+_R2"), origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
      }
}
```

Creating [`samplesheet.csv`](https://github.com/nf-core/rnaseq/blob/3.0/assets/samplesheet.csv) for the nf-core rnaseq pipeline

```r
nr <- dplyr::count(meta, group) %>% pull(n)
repl <- c(); for (i in 1:length(nr)) {repl <- c(repl, 1:nr[i])}

fqpath <- "/projects/carter-lab/MTHFR/data/RNAseq/"
  
samplesheet <- tibble(group = meta$group,
                      replicate = repl,
                      fastq_1 = paste0(fqpath, meta$sample_id, "_R1.fastq.gz"),
                      fastq_2 = paste0(fqpath, meta$sample_id, "_R2.fastq.gz"),
                      strandedness = "reverse")

write_csv(samplesheet, "output/RNAseq/1_Processing_QC/samplesheet.csv")

meta$samplesheet <- paste0(meta$group, "_R", repl)
```

## Processing fastq Files
The fastq files are processed using the [nextflow-core methylseq pipeline](https://github.com/nf-core/methylseq), and submitted to sumner with [this script](assets/RRBS_run_nf_methylseq.sbatch).  
Reads were aligned to GRCm38 reference genome as a directional library. 

```bash
cd /projects/carter-lab/MTHFR/output/RNAseq/1_Processing_QC
sbatch run_nf_rnaseq.sbatch
cp run_nf_rnaseq.sbatch /projects/carter-lab/MTHFR/assets/RNAseq_run_nf_rnaseq.sbatch
nextflow clean -f
```

## QC: Processing Pipeline
Summary reports generated by the pipeline:

```r
file.copy(from = "output/RNAseq/1_Processing_QC/nf_rnaseq_output/multiqc/star_rsem/multiqc_report.html", 
          to = "./assets/RNAseq_multiqc_report.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)

file.copy(from = "output/RNAseq/1_Processing_QC/samplesheet.csv", 
          to = "./assets/RNAseq_samplesheet.csv", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
```
1. [Overview of pipeline and output](https://nf-co.re/rnaseq/3.0/output)  
2. [QC report for all steps](assets/RNAseq_multiqc_report.html)   
3. [Sample list that matches the pipeline's notation](assets/RNAseq_samplesheet.csv)

## QC: Genotypes
Extracting genotypes from fastq files to validate metadata

```bash
#singularity pull --name ~/bin/bam-readcount shub://jaxreg.jax.org/rit-ci/bam-readcount:v0.8
#~/bin/bam-readcount $inpath/CC_18_F_R1.sorted.bam 4:148048151-148048151

dir=/projects/carter-lab/MTHFR/output/RNAseq/1_Processing_QC
ls $dir/nf_rnaseq_output/star_rsem/*[1-7].sorted.bam > $dir/bamlist.txt

singularity exec ~/bin/htslib samtools mpileup \
      -B \
      -r 4:148048141-148048161 \
      -b $dir/bamlist.txt > $dir/mp_gt.txt
```


```r
mpgt <- read_tsv("output/RNAseq/1_Processing_QC/mp_gt.txt", col_names = FALSE)
mpgt <- select(mpgt, seq(5, ncol(mpgt), 3))

#bamlist <- read_table("output/RNAseq/1_Processing_QC/bamlist.txt", col_names = FALSE)

gt_c667t <- slice(mpgt, 11) %>%
  t() %>%
  as.tibble() %>%
  mutate("A" = str_count(V1, "[a|A]"),
         "C" = str_count(V1, "[c|C]"),
         "G" = str_count(V1, "[g|G]"),
         "T" = str_count(V1, "[t|T]")) %>%
  select(-V1) %>%
  add_column(sample_id = meta$sample_id, group = meta$group, .before = 1)

gt_c667t
```

<div class="kable-table">

|sample_id |group   |  A|  C|  G|  T|
|:---------|:-------|--:|--:|--:|--:|
|34164     |CC_18_F |  0| 38|  0|  0|
|34165     |CC_18_F |  0| 22|  0|  0|
|34195     |CC_18_F |  0| 47|  0|  0|
|34196     |CC_18_F |  0| 34|  0|  0|
|38410     |CC_18_F |  0| 38|  0|  0|
|38494     |CC_18_F |  0| 30|  0|  0|
|00001     |CC_18_M |  0| 32|  0|  0|
|00002     |CC_18_M |  0| 41|  0|  0|
|00003     |CC_18_M |  0| 31|  0|  0|
|00004     |CC_18_M |  0| 44|  0|  1|
|34183     |CC_18_M |  0| 19|  0|  0|
|38490     |CC_18_M |  0| 47|  0|  0|
|08963     |CC_6_F  |  0| 43|  0|  0|
|08964     |CC_6_F  |  0| 38|  0|  0|
|08967     |CC_6_F  |  0| 27|  0|  0|
|08974     |CC_6_F  |  0| 27|  0|  0|
|48287     |CC_6_F  |  0| 32|  0|  0|
|48288     |CC_6_F  |  0| 28|  0|  0|
|08961     |CC_6_M  |  0| 39|  0|  0|
|08962     |CC_6_M  |  0| 29|  0|  0|
|08971     |CC_6_M  |  0| 36|  0|  0|
|48276     |CC_6_M  |  0| 35|  0|  0|
|48285     |CC_6_M  |  0| 28|  0|  0|
|48286     |CC_6_M  |  0| 31|  0|  0|
|34178     |CT_18_F |  0| 10|  0| 19|
|34179     |CT_18_F |  0|  7|  0| 13|
|34194     |CT_18_F |  0| 19|  0|  7|
|34198     |CT_18_F |  0| 18|  0| 18|
|38405     |CT_18_F |  0| 20|  0| 20|
|38496     |CT_18_F |  0| 24|  0| 28|
|34133     |CT_18_M |  0| 19|  0| 16|
|34134     |CT_18_M |  0|  9|  0| 16|
|34163     |CT_18_M |  0|  0|  0| 27|
|34186     |CT_18_M |  0| 10|  0| 14|
|34187     |CT_18_M |  0| 28|  0| 26|
|38403     |CT_18_M |  0| 10|  0| 15|
|08965     |CT_6_F  |  0| 16|  0| 14|
|08966     |CT_6_F  |  0| 20|  0| 19|
|08975     |CT_6_F  |  0| 15|  0| 28|
|08976     |CT_6_F  |  0| 16|  0|  8|
|48283     |CT_6_F  |  0| 16|  0| 21|
|48284     |CT_6_F  |  0| 15|  0| 12|
|08959     |CT_6_M  |  0| 22|  0| 10|
|08960     |CT_6_M  |  0| 25|  0| 18|
|48273     |CT_6_M  |  0| 15|  0|  6|
|48280     |CT_6_M  |  0| 23|  0| 23|
|48281     |CT_6_M  |  0| 17|  0| 20|
|48282     |CT_6_M  |  0| 10|  0| 16|
|34160     |TT_18_F |  0|  0|  0| 27|
|34162     |TT_18_F |  0|  0|  0| 26|
|34197     |TT_18_F |  0|  0|  0| 44|
|38406     |TT_18_F |  0|  0|  1| 29|
|38493     |TT_18_F |  0|  0|  0| 42|
|38495     |TT_18_F |  0|  0|  0| 28|
|34157     |TT_18_M |  0|  0|  0| 27|
|34158     |TT_18_M |  0|  8|  0| 19|
|34177     |TT_18_M |  0|  2|  0| 35|
|34184     |TT_18_M |  0|  0|  0| 41|
|34199     |TT_18_M |  0|  0|  0| 25|
|34200     |TT_18_M |  0|  1|  0| 27|
|08945     |TT_6_F  |  0|  0|  0| 26|
|08973     |TT_6_F  |  0|  1|  0| 20|
|48277     |TT_6_F  |  0|  0|  0| 27|
|48278     |TT_6_F  |  0|  0|  0| 28|
|48279     |TT_6_F  |  0|  1|  0| 28|
|48289     |TT_6_F  |  0|  0|  0| 40|
|55487     |TT_6_F  |  0|  0|  0| 35|
|08946     |TT_6_M  |  0|  2|  0| 34|
|08992     |TT_6_M  |  0|  0|  0| 34|
|48274     |TT_6_M  |  0|  0|  0| 42|
|48275     |TT_6_M  |  0|  0|  0| 28|
|55473     |TT_6_M  |  0|  0|  0| 26|
|55477     |TT_6_M  |  0|  0|  0| 44|

</div>



## Output
Saving

```r
saveRDS(meta, file="output/RNAseq/1_Processing_QC/samplesheet_meta.RDS")
```

Uploading to [Synapse](https://www.synapse.org/#!Synapse:syn23573590/wiki/607402) as csv file.

```r
synLogin() 
```
