---
title: "RRBS Pilot Processing and QC"
author: "Annat Haber"
date: '`r Sys.Date()`'
output:
  html_document:
    toc: true
    code_folding: hide
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(cache=TRUE,message=FALSE, warning=FALSE)
```

```{r libraries}
suppressPackageStartupMessages({
  library(methylKit)
  library(tidyverse)
  library(synapser)
})
```

```{r metadata}
meta <- readRDS("output/RRBS_Pilot/sample_meta.RDS") %>% arrange(genotype,sex, age)
```

## Preparing input files
Creating symbolic links to the original fastq files.  
Files are renamed as `<sample_id>.fastq.gz`.   
```{r, eval=FALSE}
# original files to link to
origfiles <- list.files("/projects/howell-lab/00_fastq/reagaa/RRBS/20190820_19-howell-005-run4",
                        pattern="*.fastq.gz", full.names = TRUE)

for (id in meta$sample_id) {
  flink <- paste0("./data/RRBS_Pilot/", id, "_R1.fastq.gz") # link name
  ofile <- grep(id, origfiles, value=TRUE) # original file and path
  file.symlink(ofile, flink)
}

```

## Processing fastq Files
The fastq files are processed using the [nextflow-core methylseq pipeline](https://github.com/nf-core/methylseq), and submitted to sumner with [this script](assets/RRBS_Pilot_run_nf_methylseq.sbatch).  
Reads were aligned to GRCm38 reference genome as a directional library. 
```{bash run_nf, eval=FALSE}
cd /projects/carter-lab/MTHFR/output/RRBS_Pilot/1_Processing_QC
sbatch run_nf_methylseq.sbatch
cp run_nf_methylseq.sbatch /projects/carter-lab/MTHFR/assets/RRBS_Pilot_run_nf_methylseq.sbatch
```

```{bash, eval=FALSE, include=FALSE}
singularity exec ~/bin/htslib samtools faidx \
              /projects/carter-lab/MTHFR/data/RRBS_Pilot/33139_R1.fastq.gz
cut -f1,2 33139_R1.fai > chromsizes_33139.txt

```
## QC: Processing Pipeline
Summary reports generated by the pipeline:
```{r qc_files, results='hide'}
file.copy(from = "output/RRBS_Pilot/1_Processing_QC/nf_methylseq_output/pipeline_info/results_description.html", 
          to = "./assets/RRBS_Pilot_results_description.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
file.copy(from = "output/RRBS_Pilot/1_Processing_QC/nf_methylseq_output/MultiQC/multiqc_report.html", 
          to = "./assets/RRBS_Pilot_multiqc_report.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
```
1. [Overview of pipeline and output](assets/RRBS_Pilot_results_description.html)  
2. [QC report for all steps](assets/RRBS_Pilot_multiqc_report.html)  

## Generating methylation objects 
Methylation calls are extracted from the coverage files generated by bismark.  
```{r meth_cov}
covfiles <- list.files("output/RRBS_Pilot/1_Processing_QC/nf_methylseq_output/bismark_methylation_calls/methylation_coverage", full.names = TRUE)
names(covfiles) <- gsub("(\\d+)_R1_trimmed_bismark_bt2.bismark.cov.gz", "\\1", basename(covfiles))
covfiles <- as.list(covfiles[meta$sample_id])
```

Sites with less than 10x read coverage are filtered out.  
Only sites within CpG context that are covered in all samples are retained for subsequent analyses.
```{r raw_meth, cache=FALSE}
library(tidyverse)
library(methylKit)

raw_meth <- methRead(covfiles,
           sample.id=as.list(names(covfiles)),
           assembly="mm10.p6",
           treatment=meta$treatment,
           context="CpG",
           pipeline='bismarkCoverage',
           mincov = 10
           ) %>%
  filterByCoverage(lo.count=10, lo.perc=NULL, hi.count=NULL, hi.perc=99.9)
```

```{r allMeth}
united_meth <-  methylKit::unite(raw_meth, destrand=FALSE)
```

## QC: Methylation distribution and read coverage {.tabset .tabset-fade .tabset-pills}
Distributions are largely the same for all samples.  
The vast majority of sites have either 0% or 100% methylation, which is what we want.  
Coverage is highly right-skewed, which tells us that we don't need to worry about PCR duplication bias. A bias would have been indicated as an additional peak on the right.

```{r, qc, results='asis', out.width=c('50%', '50%')}
for(i in 1:nrow(meta)){
  cat("### ", paste(meta$sample_id[i], meta$genotype[i], meta$sex[i], meta$age[i], sep="_"),"\n")
  cat(getMethylationStats(raw_meth[[i]], plot=TRUE, both.strands=FALSE))
  cat(getCoverageStats(raw_meth[[i]], plot=TRUE, both.strands=FALSE))
  cat('\n\n')
}
rm(raw_meth)
```

## Output

Calculating percent methylation per site per sample (samples x sites matrix).
```{r perc.meth}
site_meth <- t(percMethylation(united_meth))
# Adding positions as column names:
colnames(site_meth) <- paste(united_meth$chr, united_meth$start, sep="_")
```

Saving methylation objects as RDS for subsequent analyses.
```{r output}
saveRDS(united_meth, file="output/RRBS_Pilot/1_Processing_QC/united_meth.RDS")
saveRDS(site_meth, file="output/RRBS_Pilot/1_Processing_QC/persite_meth.RDS")
write_csv(data.frame(site_meth), file="output/RRBS_Pilot/1_Processing_QC/RRBS_Pilot_persite_meth.csv")

```

Uploading percent methylation per site to [Synapse](https://www.synapse.org/#!Synapse:syn23573590/wiki/607402) as csv file.
```{r synapse, results='hide'}
synLogin() 

file <- File("output/RRBS_Pilot/1_Processing_QC/RRBS_Pilot_persite_meth.csv",
             contentType = "text/plain",
             description = "RRBS_Pilot Percent methylation per site per sample",
             parent = "syn25174488")
provenance <- Activity(used = "syn25175205",
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_Pilot_1_Processing_QC.Rmd")
file <- synStore(file, activity=provenance)

```