---
title: "RRBS Processing and QC"
author: "Annat Haber"
date: '`r Sys.Date()`'
output:
  html_document:
    toc: true
    code_folding: hide
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = TRUE)
```

```{r libraries}
suppressPackageStartupMessages({
  library(synapser)
  library(methylKit)
  library(tidyverse)
})
```

```{r metadata}
meta <- readRDS("output/RRBS/sample_meta.RDS") %>% arrange(genotype,sex, age)
```

## Linking input fastq files
Creating symbolic links to the original fastq files.  
Files are renamed as `<sample_id>.fastq.gz`. Sample ids are padded with zeros to have five digits      
```{r symlink_data, eval=FALSE}
# original files to link to
origfiles <- c(list.files("/projects/howell-lab/00_fastq/reagaa/RRBS/20200909_20-howell-004-run2",
                        pattern=".*GT20.*.fastq.gz", full.names = TRUE),
               list.files("/projects/howell-lab/00_fastq/reagaa/RRBS/20200909_20-howell-004-run3",
                        pattern=".*GT20.*.fastq.gz", full.names = TRUE))

for (id in meta$tag_ids) {
  if (id=="48289") { ## typo in the fastq file name
      flink <- paste0("data/RRBS/", str_pad(id, 5, pad = "0"), "_R1.fastq.gz") # link name and path
      ofile <- grep("/49289_GT20", origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
  } else {
      flink <- paste0("data/RRBS/", str_pad(id, 5, pad = "0"), "_R1.fastq.gz") # link name and path
      ofile <- grep(paste0("/",id,"_GT20"), origfiles, value=TRUE) # original file name and path
      file.symlink(ofile, flink)
  }
}

```

## Processing fastq Files
The fastq files are processed using the [nextflow-core methylseq pipeline](https://github.com/nf-core/methylseq), and submitted to sumner with [this script](assets/RRBS_run_nf_methylseq.sbatch).  
Reads were aligned to GRCm38 reference genome as a directional library. 
```{bash run_nf, eval=FALSE}
cd /projects/carter-lab/MTHFR/output/RRBS/1_Processing_QC
sbatch run_nf_methylseq.sbatch
cp run_nf_methylseq.sbatch /projects/carter-lab/MTHFR/assets/RRBS_run_nf_methylseq.sbatch
nextflow clean -f
```

## QC: Processing Pipeline
Summary reports generated by the pipeline:
```{r qc_files, results='hide'}
file.copy(from = "output/RRBS/1_Processing_QC/nf_methylseq_output/pipeline_info/results_description.html", 
          to = "./assets/RRBS_results_description.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
file.copy(from = "output/RRBS/1_Processing_QC/nf_methylseq_output/MultiQC/multiqc_report.html", 
          to = "./assets/RRBS_multiqc_report.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
```
1. [Overview of pipeline and output](assets/RRBS_results_description.html)  
2. [QC report for all steps](assets/RRBS_multiqc_report.html)   

## Generating methylation objects 
Methylation calls are extracted from the coverage files generated by bismark.  
```{r meth_cov}
covfiles <- list.files("output/RRBS/1_Processing_QC/nf_methylseq_output/bismark_methylation_calls/methylation_coverage", full.names = TRUE)
names(covfiles) <- gsub("(\\d+)_R1_trimmed_bismark_bt2.bismark.cov.gz", "\\1", basename(covfiles))
covfiles <- as.list(covfiles[meta$sample_id])
```
Sites with less than 10x read coverage are filtered out.  
Only sites within CpG context that are covered in all samples are retained for subsequent analyses.
```{r raw_meth, cache=FALSE}
library(tidyverse)
library(methylKit)

raw_meth <- methRead(covfiles,
           sample.id=as.list(names(covfiles)),
           assembly="mm10.p6",
           treatment=meta$treatment,
           context="CpG",
           pipeline='bismarkCoverage',
           mincov = 10
           ) %>%
  filterByCoverage(lo.count=10, lo.perc=NULL, hi.count=NULL, hi.perc=99.9)
```

```{r allMeth, cache=FALSE}
#raw_meth <- readRDS("output/RRBS/1_Processing_QC/raw_meth.RDS")
united_meth <-  methylKit::unite(raw_meth, destrand=FALSE)
```

Calculating percent methylation per site per sample (samples x sites matrix).
```{r perc.meth}
site_meth <- t(percMethylation(united_meth))
# Adding positions as column names:
colnames(site_meth) <- paste(united_meth$chr, united_meth$start, sep="_")
```

## QC: Methylation distribution and read coverage {.tabset .tabset-fade .tabset-pills}
Distributions are largely the same for all samples.  
The vast majority of sites have either 0% or 100% methylation, which is what we want.  
Coverage is highly right-skewed, which tells us that we don't need to worry about PCR duplication bias. A bias would have been indicated as an additional peak on the right.
```{r, qc, results='asis', out.width=c('50%', '50%'), cache=TRUE}
for(i in 1:nrow(meta)){
  cat("### ", paste(meta$sample_id[i], meta$genotype[i], meta$sex[i], meta$age[i], sep="_"),"\n")
  cat(getMethylationStats(raw_meth[[i]], plot=TRUE, both.strands=FALSE))
  cat(getCoverageStats(raw_meth[[i]], plot=TRUE, both.strands=FALSE))
  cat('\n\n')
}
rm(raw_meth)
```

## Output
Saving methylation objects for subsequent analyses.
```{r output}
saveRDS(united_meth, file="output/RRBS/1_Processing_QC/united_meth.RDS")
saveRDS(site_meth, file="output/RRBS/1_Processing_QC/persite_meth.RDS")
write_csv(data.frame(site_meth), file="output/RRBS/1_Processing_QC/RRBS_persite_meth.csv")
```

Uploading to [Synapse](https://www.synapse.org/#!Synapse:syn25174486) percent methylation per site as csv file and the raw methylKit object (fitered and merged).
```{r synapse, results='hide'}
synLogin() 

file <- File("output/RRBS/1_Processing_QC/united_meth.RDS",
             contentType = "text/plain",
             description = "RRBS raw methylKit; filtered for coverage x10 and including only sites that are covered in all samples",
             parent = "syn25174486")
provenance <- Activity(used = "syn25174517",
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_1_Processing_QC.Rmd")
file <- synStore(file, activity=provenance)

file <- File("output/RRBS/1_Processing_QC/RRBS_persite_meth.csv",
             contentType = "text/plain",
             description = "RRBS Percent methylation per site per sample",
             parent = "syn25174486")
provenance <- Activity(used = "syn25174517",
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_1_Processing_QC.Rmd")
file <- synStore(file, activity=provenance)

```
