---
title: "RRBS Differential Methylation"
subtitle: "Separated by genotype, age, and sex"
author: "Annat Haber"
date: '`r Sys.Date()`'
output: 
  html_document:
    toc: true
    toc_depth: 3
    code_folding: hide
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(cache = TRUE, 
                      #cache.path = "output/RRBS/3_Differential_Methylation",
                      message=FALSE, warning=FALSE)
```

```{r libraries}
rm(list=ls())

suppressPackageStartupMessages({
  library(methylKit)
  library(synapser)
  library(tidyverse)
  library(GenomicRanges)
  library(genomation)
  library(ChIPpeakAnno)
  library(clusterProfiler)
  library(GOsummaries)
  library(org.Mm.eg.db)
  library(biomaRt)
  library(vcd)
})

```

## Input
Read sample metadata and methylation calls.  
Genotypes were already switched for two samples (34158 from TT to CT; 34163 from CT to TT)
```{r input_project}
united_meth <- readRDS(file="output/RRBS/1_Processing_QC/united_meth.RDS")
meta <- read_csv(file="output/RRBS/2_Exploratory_Analysis/RRBS_globalmeth_hmc.csv",
                 col_types = cols(.default = "c"))
```

```{r input_synapse, eval=FALSE}
synLogin() 

united_meth <- readRDS(synGet("syn25834047")$path)
meta <- read_csv(synGet("syn25834048")$path,
                 col_types = cols(.default = "c"))
```

## differentially Methylated CpG sites  {.tabset .tabset-fade .tabset-pills}
Finding differentially methylated CpG sites separately for age groups and sex.  
Mapping to known genes by finding overlap between the snp's and mm10 reference.  
The reference annotation was downloaded directly from http://genome.ucsc.edu/cgi-bin/hgTables 
```{r refseq_gr}
# read reference and convert to GRanges
refseq <- read_tsv("data/mm10.refseq.genes.txt") 

ref.gr <- GRanges(
        seqnames = substr(refseq$chrom, 4,5),
        ranges=IRanges(start = refseq$txStart, end = refseq$txEnd),
        mgi_name=refseq$name2
        )
names(ref.gr) <- refseq$name
rm(refseq)
```

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
For each combination of sex (M,F) and age (6,18) we compare CT to CC and TT to CC.
Top genes are identified as significant at qvalue < 0.01 and methylation difference >10
```{r diffM_obj_gt}
grp <- meta %>%
  filter(genotype!="CC") %>%
  distinct(genotype, sex, age)

diffML_gt <- sig.snpL_gt <- list()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt <- grp$genotype[i]
  sx <- grp$sex[i]
  ag <- grp$age[i]
  gr <- paste(gt, sx, ag, sep="_")
  
  # get sample_id and grouping factor (genotype) for this comparison
  cov <- meta %>%
    filter(genotype %in% c("CC",gt) & sex == sx & age == ag) %>%
    mutate(genotype = ifelse(genotype==gt, 1, 0)) %>%
    dplyr::select(sample_id, genotype, sex, age)
  
  # DiffMeth object and analysis
  diffM <- reorganize(united_meth, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$genotype) %>%
            calculateDiffMeth()
  
  # extract significant sites and convert to GRanges
  snp.gr <- getMethylDiff(diffM, difference = 10, qvalue=0.01, type="all") %>%
    as("GRanges")
  names(snp.gr) <- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=":")

  # Find overlap with reference
  overlap <- findOverlaps(snp.gr, ref.gr)
  hits <- ref.gr$mgi_name[slot(overlap, "to")]
  names(hits) <- names(snp.gr)[slot(overlap, "from")]
  hits <- hits[!duplicated(hits)]
  snpDMgenes <- snp.gr[names(hits),]
  mcols(snpDMgenes)$mgi_name <- hits
  
  snpDMgenes <- data.frame(snpDMgenes) %>%
    dplyr::select(chromosome=seqnames, everything()) %>%
    arrange(chromosome, meth.diff)
  
  diffML_gt[[gr]] <- diffM
  sig.snpL_gt[[gr]] <- snpDMgenes
  
  rm(gt, sx, ag, cov, diffM, snp.gr, overlap, hits, snpDMgenes)
}
```

```{r, diffM_plot_gt, message=FALSE, warning=FALSE, fig.height = 6, results='asis'}
for(gr in names(diffML_gt)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_gt[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n")
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)
  cat('\n\n')
}

rm(diffM)
```

### Old vs Young {.tabset .tabset-fade .tabset-pills}
For each combination of sex (M,F) and genotype (CC, CT, TT) we compare old to young samples (18 vs 6 months).  
Top genes are identified as significant at qvalue < 0.01 and methylation difference >10
```{r diffM_obj_age}
grp <- meta %>%
  filter(age=="18") %>%
  distinct(genotype, sex, age)

diffML_age <- sig.snpL_age <- list()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt <- grp$genotype[i]
  sx <- grp$sex[i]
  ag <- grp$age[i]
  gr <- paste(gt, sx, ag, sep="_")
  
  # get sample_id and grouping factor (genotype) for this comparison
  cov <- meta %>%
    filter(sex == sx & genotype == gt) %>%
    mutate(age = ifelse(age=="18", 1, 0)) %>%
    dplyr::select(sample_id, genotype, sex, age)
  
  # DiffMeth object and analysis
  diffM <- reorganize(united_meth, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$age) %>%
            calculateDiffMeth()
  
  # extract significant sites and convert to GRanges
  snp.gr <- getMethylDiff(diffM, difference = 10, qvalue=0.01, type="all") %>%
    as("GRanges")
  names(snp.gr) <- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=":")

  # Find overlap with reference
  overlap <- findOverlaps(snp.gr, ref.gr)
  hits <- ref.gr$mgi_name[slot(overlap, "to")]
  names(hits) <- names(snp.gr)[slot(overlap, "from")]
  hits <- hits[!duplicated(hits)]
  snpDMgenes <- snp.gr[names(hits),]
  mcols(snpDMgenes)$mgi_name <- hits
  
  snpDMgenes <- data.frame(snpDMgenes) %>%
    dplyr::select(chromosome=seqnames, everything()) %>%
    arrange(chromosome, meth.diff)
  
  diffML_age[[gr]] <- diffM
  sig.snpL_age[[gr]] <- snpDMgenes
  
  rm(gt, sx, ag, cov, diffM, snp.gr, overlap, hits, snpDMgenes)
}
```

```{r, diffM_plot_age, message=FALSE, warning=FALSE, fig.height = 6, results='asis'}
for(gr in names(diffML_age)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_age[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "18", "6")), "\n")
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)
  cat('\n\n')
}

rm(diffM)
```

## Gene parts annotation {.tabset .tabset-fade .tabset-pills}
Gene parts distribution for differentially-methylated sites, defines with qvalue < 0.01 and methylation difference >10
```{r ref_anno}
gene.obj=readTranscriptFeatures("data/mm10.refseq.genes.bed")
cpg.obj=readFeatureFlank("data/mm10.refseq.genes.bed",
                           feature.flank.name=c("CpGi","shores"))
```

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
```{r, gene_parts_gt, message=FALSE, warning=FALSE, results='asis', out.width=c('50%', '50%')}
for(gr in names(diffML_gt)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_gt[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n", "\n")
  # hypermethylated sites with difference > 10%
  diffM10.hyper <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hyper") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hyper,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hyper-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hyper-methylated flank annotation; > 10% diff.")


  # hypomethylated sites with difference > 10%
  diffM10.hypo <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hypo") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hypo,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hypo-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hypo-methylated flank annotation; > 10% diff.")
  
  cat('\n\n')
}

```

### Old vs Young {.tabset .tabset-fade .tabset-pills}
```{r, gene_parts_age, message=FALSE, warning=FALSE, results='asis', out.width=c('50%', '50%')}
for(gr in names(diffML_age)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_age[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "18", "6")), "\n", "\n")
  # hypermethylated sites with difference > 10%
  diffM10.hyper <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hyper") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hyper,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hyper-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hyper-methylated flank annotation; > 10% diff.")


  # hypomethylated sites with difference > 10%
  diffM10.hypo <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hypo") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hypo,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hypo-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hypo-methylated flank annotation; > 10% diff.")
  
  cat('\n\n')
}

```

## KEGG Annotation {.tabset .tabset-fade .tabset-pills}

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
```{r, kegg_gt, results="hide"}
keggL_gt <- list()

for (gr in names(sig.snpL_gt)) {
  
  ## hyper-methylated genes
  sig.gs <- sig.snpL_gt[[gr]] %>%
    dplyr::filter(meth.diff > 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_gt[[gr]][["hyper"]] <- anno

  ## hypo-methylated genes
  sig.gs <- sig.snpL_gt[[gr]] %>%
    dplyr::filter(meth.diff < 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_gt[[gr]][["hypo"]] <- anno
}

rm(sig.gs, sig.gs.ez, anno)
```

```{r, kegg_plot_gt, results="asis", fig.width = 10}
for(gr in names(keggL_gt)){
  cat("#### ", gr ,"\n")
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n", "\n")

  anno <- keggL_gt[[gr]][["hyper"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hyper-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hyper")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_gt_",gr,".csv"))

  anno <- keggL_gt[[gr]][["hypo"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hypo-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hypo")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_gt_",gr,".csv"), append = TRUE)
  
  cat('\n\n')

}

rm(anno)
```

### Old vs Young {.tabset .tabset-fade .tabset-pills}
```{r, kegg_age, results="hide"}
keggL_age <- list()

for (gr in names(sig.snpL_age)) {
  
  ## hyper-methylated genes
  sig.gs <- sig.snpL_age[[gr]] %>%
    dplyr::filter(meth.diff > 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_age[[gr]][["hyper"]] <- anno

  ## hypo-methylated genes
  sig.gs <- sig.snpL_age[[gr]] %>%
    dplyr::filter(meth.diff < 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_age[[gr]][["hypo"]] <- anno
}

rm(sig.gs, sig.gs.ez, anno)

```

```{r, kegg_plot_age, results="asis", fig.width = 10}
for(gr in names(keggL_age)){
  cat("#### ", gr ,"\n")
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "18", "6")), "\n", "\n")

  anno <- keggL_age[[gr]][["hyper"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hyper-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hyper")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_age_",gr,".csv"))

  anno <- keggL_age[[gr]][["hypo"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hypo-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hypo")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_age_",gr,".csv"), append = TRUE)
  
  cat('\n\n')

}

rm(anno)
```

## Cell-specificity {.tabset .tabset-fade .tabset-pills}
Cell type markers in the brain are taken from McKenzie et al. (Supplemental File 1, DOI:10.1038/s41598-018-27293-5).
Cell specificity for a given gene was determined in this study as minimum log fold change when pairwise comparing the expression of the gene in the given cell type with the expression of the gene in all other cell types. Cell type for that gene was then determined as the cell type with the highest specificity. The top 1000 genes were included in the final dataset as markers for each cell type.

Here I'm using cell specificity score to look for overlap between cell-specific genes and hypo/hyper methylated genes. I use chi-square to test for biased representations of any cell type genes in either the hypo- or hyper-methylated genes, and a plot of chi-square deviations from expectation for visualization.  

```{r cell_spec_ref, eval=FALSE, include=FALSE}
cell.types <- c("Astroctyes", "Endothelial", "Microglia", "Neurons", "Oligodendrocytes")
sets <- c("hypo","hyper")

# read cell-specificity reference
hum.specific <- read.csv("data/McKenzie_2017_Brain_Cell_Specific_Markers_Human_Specificity.csv", header=T, stringsAsFactors=F)
hum.specific <- split(hum.specific, hum.specific$Celltype, drop=TRUE) 
names(hum.specific) <- cell.types

```

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
```{r, cell_spec_gt, results="asis", eval=FALSE, include=FALSE}

for (gr in names(sig.snpL_gt)) {
  cat("#### ", gr ,"\n")
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n", "\n")
  sig.gs <- sig.snpL_gt[[gr]]

# get gene sets
gene.sets <- lapply(sig.gs, function(l){dplyr::select(l, mgi_name)})

gene.sets.hg <- list()
for (s in names(sig.gs)) {
    gene.sets.hg[[s]] = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", 
                              values = sig.gs[[s]]$mgi_name, mart = mouse,
                              attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)$HGNC.symbol

}

# Number of marker genes for each cell type in each gene set
specific.set <- sapply(sets, function(s) {
  sapply(cell.types, function(ct) length(intersect(hum.specific[[ct]]$gene, gene.sets.hg[[s]])), USE.NAMES=T)
})

print(specific.set)
chisq.test(specific.set)
assoc(t(specific.set), shade=TRUE,labeling_args = list(abbreviate_labs = c(6, 9)))

}
rm(sig.gs, gene.sets, gene.sets.hg, specific.set)

```

## Output
Saving lists of significantly hyper/hypo methylated genes to project folder
```{r output_save}
# variant vs wt
write_csv(as.tibble(t(c(colnames(sig.snpL_gt[[1]]), "comparison"))), 
            file="output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv",
          col_names = F)

for (gr in names(sig.snpL_gt)) {
  sig.snpL_gt[[gr]] %>%
    dplyr::mutate(comparison=gr) %>%
    write_csv(file="output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv", append=TRUE)
  }

# age 18 vs 6
write_csv(as.tibble(t(c(colnames(sig.snpL_age[[1]]), "comparison"))), 
            file="output/RRBS/3_Differential_Methylation/sig_dmg_age.csv",
          col_names = F)

for (gr in names(sig.snpL_age)) {
  sig.snpL_age[[gr]] %>%
    dplyr::mutate(comparison=gr) %>%
    write_csv(file="output/RRBS/3_Differential_Methylation/sig_dmg_age.csv", append=TRUE)
  }

```

Uploading lists of significant genes to [synapse]((https://www.synapse.org/#!Synapse:syn25174486)
```{r output_synapse, results='hide'}
synLogin() 
## variant vs wt
file <- File("output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv",
             contentType = "text/plain",
             description = "lists of significantly hyper/hypo methylated genes; variant vs wt",
             parent = "syn25174486")

provenance <- Activity(used = c("syn25834048", "syn25834047"),
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_3_Differential_Methylation_gtsxag.Rmd")

file <- synStore(file, activity=provenance)

# 18 vs 6 months
file <- File("output/RRBS/3_Differential_Methylation/sig_dmg_age.csv",
             contentType = "text/plain",
             description = "lists of significantly hyper/hypo methylated genes; 18 vs 6 months",
             parent = "syn25174486")

provenance <- Activity(used = c("syn25834048", "syn25834047"),
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_3_Differential_Methylation_gtsxag.Rmd")

file <- synStore(file, activity=provenance)

```
