---
title: "RRBS Pilot Processing and QC"
author: "Annat Haber"
date: '2021-05-19'
output:
  html_document:
    toc: true
    code_folding: hide
---


```r
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(cache=TRUE,message=FALSE, warning=FALSE)
```


```r
suppressPackageStartupMessages({
  library(methylKit)
  library(tidyverse)
  library(synapser)
})
```


```r
meta <- readRDS("output/RRBS_Pilot/sample_meta.RDS") %>% arrange(genotype,sex, age)
```

## Preparing input files
Creating symbolic links to the original fastq files.  
Files are renamed as `<sample_id>.fastq.gz`.   

```r
# original files to link to
origfiles <- list.files("/projects/howell-lab/00_fastq/reagaa/RRBS/20190820_19-howell-005-run4",
                        pattern="*.fastq.gz", full.names = TRUE)

for (id in meta$sample_id) {
  flink <- paste0("./data/RRBS_Pilot/", id, "_R1.fastq.gz") # link name
  ofile <- grep(id, origfiles, value=TRUE) # original file and path
  file.symlink(ofile, flink)
}
```

## Processing fastq Files
The fastq files are processed using the [nextflow-core methylseq pipeline](https://github.com/nf-core/methylseq), and submitted to sumner with [this script](assets/RRBS_Pilot_run_nf_methylseq.sbatch).  
Reads were aligned to GRCm38 reference genome as a directional library. 

```bash
cd /projects/carter-lab/MTHFR/output/RRBS_Pilot/1_Processing_QC
sbatch run_nf_methylseq.sbatch
cp run_nf_methylseq.sbatch /projects/carter-lab/MTHFR/assets/RRBS_Pilot_run_nf_methylseq.sbatch
```


## QC: Processing Pipeline
Summary reports generated by the pipeline:

```r
file.copy(from = "output/RRBS_Pilot/1_Processing_QC/nf_methylseq_output/pipeline_info/results_description.html", 
          to = "./assets/RRBS_Pilot_results_description.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
file.copy(from = "output/RRBS_Pilot/1_Processing_QC/nf_methylseq_output/MultiQC/multiqc_report.html", 
          to = "./assets/RRBS_Pilot_multiqc_report.html", 
          overwrite = TRUE, recursive = FALSE, copy.mode = TRUE, copy.date = TRUE)
```
1. [Overview of pipeline and output](assets/RRBS_Pilot_results_description.html)  
2. [QC report for all steps](assets/RRBS_Pilot_multiqc_report.html)  

## Generating methylation objects 
Methylation calls are extracted from the coverage files generated by bismark.  

```r
covfiles <- list.files("output/RRBS_Pilot/1_Processing_QC/nf_methylseq_output/bismark_methylation_calls/methylation_coverage", full.names = TRUE)
names(covfiles) <- gsub("(\\d+)_R1_trimmed_bismark_bt2.bismark.cov.gz", "\\1", basename(covfiles))
covfiles <- as.list(covfiles[meta$sample_id])
```

Sites with less than 10x read coverage are filtered out.  
Only sites within CpG context that are covered in all samples are retained for subsequent analyses.

```r
library(tidyverse)
library(methylKit)

raw_meth <- methRead(covfiles,
           sample.id=as.list(names(covfiles)),
           assembly="mm10.p6",
           treatment=meta$treatment,
           context="CpG",
           pipeline='bismarkCoverage',
           mincov = 10
           ) %>%
  filterByCoverage(lo.count=10, lo.perc=NULL, hi.count=NULL, hi.perc=99.9)
```


```r
united_meth <-  methylKit::unite(raw_meth, destrand=FALSE)
```

## QC: Methylation distribution and read coverage {.tabset .tabset-fade .tabset-pills}
Distributions are largely the same for all samples.  
The vast majority of sites have either 0% or 100% methylation, which is what we want.  
Coverage is highly right-skewed, which tells us that we don't need to worry about PCR duplication bias. A bias would have been indicated as an additional peak on the right.


```r
for(i in 1:nrow(meta)){
  cat("### ", paste(meta$sample_id[i], meta$genotype[i], meta$sex[i], meta$age[i], sep="_"),"\n")
  cat(getMethylationStats(raw_meth[[i]], plot=TRUE, both.strands=FALSE))
  cat(getCoverageStats(raw_meth[[i]], plot=TRUE, both.strands=FALSE))
  cat('\n\n')
}
```

###  33140_CC_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-1.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-2.png" width="50%" />

###  34036_CC_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-3.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-4.png" width="50%" />

###  35300_CC_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-5.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-6.png" width="50%" />

###  43763_CC_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-7.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-8.png" width="50%" />

###  43764_CC_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-9.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-10.png" width="50%" />

###  43765_CC_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-11.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-12.png" width="50%" />

###  44292_CC_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-13.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-14.png" width="50%" />

###  33139_CT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-15.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-16.png" width="50%" />

###  33174_CT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-17.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-18.png" width="50%" />

###  33193_CT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-19.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-20.png" width="50%" />

###  34035_CT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-21.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-22.png" width="50%" />

###  43768_CT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-23.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-24.png" width="50%" />

###  44275_CT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-25.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-26.png" width="50%" />

###  44276_CT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-27.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-28.png" width="50%" />

###  44277_CT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-29.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-30.png" width="50%" />

###  33178_TT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-31.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-32.png" width="50%" />

###  34004_TT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-33.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-34.png" width="50%" />

###  34014_TT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-35.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-36.png" width="50%" />

###  34027_TT_F_13 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-37.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-38.png" width="50%" />

###  38424_TT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-39.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-40.png" width="50%" />

###  38425_TT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-41.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-42.png" width="50%" />

###  38444_TT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-43.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-44.png" width="50%" />

###  44278_TT_F_4 
<img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-45.png" width="50%" /><img src="RRBS_Pilot_1_Processing_QC_files/figure-html/qc-46.png" width="50%" />

```r
rm(raw_meth)
```

## Output

Calculating percent methylation per site per sample (samples x sites matrix).

```r
site_meth <- t(percMethylation(united_meth))
# Adding positions as column names:
colnames(site_meth) <- paste(united_meth$chr, united_meth$start, sep="_")
```

Saving methylation objects as RDS for subsequent analyses.

```r
saveRDS(united_meth, file="output/RRBS_Pilot/1_Processing_QC/united_meth.RDS")
saveRDS(site_meth, file="output/RRBS_Pilot/1_Processing_QC/persite_meth.RDS")
write_csv(data.frame(site_meth), file="output/RRBS_Pilot/1_Processing_QC/RRBS_Pilot_persite_meth.csv")
```

Uploading percent methylation per site to [Synapse](https://www.synapse.org/#!Synapse:syn23573590/wiki/607402) as csv file.

```r
synLogin() 

file <- File("output/RRBS_Pilot/1_Processing_QC/RRBS_Pilot_persite_meth.csv",
             contentType = "text/plain",
             description = "RRBS_Pilot Percent methylation per site per sample",
             parent = "syn25174488")
provenance <- Activity(used = "syn25175205",
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_Pilot_1_Processing_QC.Rmd")
file <- synStore(file, activity=provenance)
```
