---
title: "RRBS Pilot Differential Methylation Annotation"
author: "Annat Haber"
date: '`r Sys.Date()`'
output: 
  html_document:
    toc: true
    toc_depth: 3
    code_folding: hide
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = FALSE, cache.path = "output/RRBS_Pilot/3_Differential_Methylation", eval=FALSE)
```

```{r libraries}
rm(list=ls())

suppressPackageStartupMessages({
  library(methylKit)
  library(tidyverse)

  library(GenomicRanges)
  library(genomation)
  library(ChIPpeakAnno)
  library(clusterProfiler)
  library(GOsummaries)
  library(org.Mm.eg.db)
})

diffM <- readRDS("output/2_Analysis/diffM.TT.RDS")
```

## Gene parts annotation
```{r, gene_part_annotation, results="hold"}
# Get hypermethylated sites with difference > 10% and convert to GRanges
diffM10.hyper <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hyper") %>%
  as("GRanges")

# Get hypomethylated sites with difference > 10% and convert to GRanges
diffM10.hypo <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hypo") %>%
  as("GRanges")


# Get gene parts annotation
gene.obj=readTranscriptFeatures("data/mm10.refseq.genes.bed0.txt")

hypo10.ann <- annotateWithGeneParts(diffM10.hypo,  gene.obj)
plotTargetAnnotation(hypo10.ann, precedence=TRUE,
    main="Hypo-methylated gene parts; > 10% diff.")

hyper10.ann <- annotateWithGeneParts(diffM10.hyper, gene.obj)
plotTargetAnnotation(hyper10.ann, precedence=TRUE,
    main="Hyper-methylated gene parts; > 10% diff.")

# Get flank annotation
cpg.obj=readFeatureFlank("data/mm10.refseq.genes.bed0.txt",
                           feature.flank.name=c("CpGi","shores"))

diffCpGann.hypo10 <- annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                         feature.name="CpGi",flank.name="shores")
plotTargetAnnotation(diffCpGann.hypo10,col=c("green","gray","white"),
       main="Hypo-methylated flank annotation; > 10% diff.")


diffCpGann.hyper10 <- annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                         feature.name="CpGi",flank.name="shores")
plotTargetAnnotation(diffCpGann.hyper10,col=c("green","gray","white"),
       main="Hyper-methylated flank annotation; > 10% diff.")

```

## Map snp's to known genes 
Using MGI names
```{r, map_DM_genes}
snpDM <- getMethylDiff(diffM, difference = 0, qvalue=0.01, type="all") 
chr <- substr(snpDM@.Data[[1]], 4, 5)
snpDM@.Data[[1]] <- chr
snp.gr <- as(snpDM, "GRanges")
names(snp.gr) <- paste(chr, snpDM@.Data[[2]], sep=":")

# downloaded reference annotation directly from http://genome.ucsc.edu/cgi-bin/hgTables
# removed X and Y chromosomes
refseq <- read_tsv("data/mm10.refseq.genes.txt") 
chr <- substr(refseq$chrom, 4,5)
ref.gr <- GRanges(
        seqnames = chr,
        ranges=IRanges(start = refseq$txStart, end = refseq$txEnd),
        mgi_name=refseq$name2
        )
names(ref.gr) <- refseq$name

overlap <- findOverlaps(snp.gr, ref.gr)
hits <- ref.gr$mgi_name[slot(overlap, "to")]
names(hits) <- names(snp.gr)[slot(overlap, "from")]
hits <- hits[!duplicated(hits)]
snpDMgenes <- snp.gr[names(hits),]
mcols(snpDMgenes)$mgi_name <- hits

write.table(data.frame(snpDMgenes), file = "output/3_Annotation/snpDMgenes.csv", sep=",", col.names=TRUE, row.names=FALSE, quote=FALSE)

#### Extract significant genes and convert to entrez id
sig.gs <- list("hypo"=
                 (data.frame(snpDMgenes) %>%
                    filter(qvalue < 0.01 & meth.diff < -10) %>%
                    arrange(meth.diff)), 
               "hyper"=
                 (data.frame(snpDMgenes) %>%
                    filter(qvalue < 0.01 & meth.diff > 10) %>%
                    arrange(-meth.diff)))
  
# get entrez id
sig.gs.ez <- list()
for (gs in names(sig.gs)) {
  sig.gs.ez[[gs]] <- bitr(sig.gs[[gs]]$mgi_name, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
}

```

## Annotate mapped genes
### KEGG annotation
```{r, kegg_annotation, fig.width = 10}
#library(enrichplot)
### KEGG
kegg_anno <- list()
for (gs in names(sig.gs.ez)) {
  cat("\n Annotating:", gs,"-methylated genes\n")
  
  anno <- enrichKEGG(gene = sig.gs.ez[[gs]],
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for ", gs, "-methylated sites; >10%")))
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  anno <- as.data.frame(anno@result)
  write_csv(anno, path=paste0("output/3_Annotation/kegg_anno_", gs, ".csv"))

  kegg_anno[[gs]] <- anno

}

saveRDS(kegg_anno, file="output/3_Annotation/kegg_anno.RDS")


```
  
### GO annotation 
Doesn't work; not enough enriched terms
```{r, GO_annotation, eval=FALSE}
goBP_anno <- list()
for (gs in names(sig.gs.ez)) {
  cat("\n Annotating:", gs,"-methylated genes\n")
  
  go_anno <- enrichGO(gene = sig.gs.ez[[gs]],
                OrgDb         = "org.Mm.eg.db",
                keyType       = 'ENTREZID',
                ont           = "BP",  # Biological processes
                pAdjustMethod = "bonferroni",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
                readable=TRUE)
  
  go_anno <- as.data.frame(go_anno@result)
  colnames(go_anno)[1] <- "GO.ID"
  colnames(go_anno)[8] <- "mgi_name"

  goBP_anno[[gs]] <- go_anno

}

saveRDS(goBP_anno, file="output/3_Annotation/goBP_anno.RDS")


## GOsummaries and cloud plots
library(GOsummaries)

gosum <- list()
for (gs in names(goBP_anno)) {
  gosum[[gs]] <- gosummaries(goBP_anno[[gs]]$mgi_name, ordered_query=FALSE, max_p_value = 0.5) %>%
    add_to_slot.gosummaries("Title", list(paste0(gs,"-methylated; >10%")))
}

for (gs in names(gosum)) {
  plot.gosummaries(gosum[[gs]],filename=paste0("output/3_Annotation/GOcloud_",gs,".pdf"))
}
###################
```



```{r, eval=FALSE, include=FALSE}
### Generate tables for FUMA
# Generate a dataframe to filter from
diffM.df <- as.data.frame(diffM@.Data)
colnames(diffM.df) <- diffM@names
diffM.df <-  dplyr::select(diffM.df, CHR=chr, BP=start, P=qvalue, M=meth.diff)
diffM.df$CHR <- substr(diffM.df$CHR,4,4)

# Filter significantly hypo- and hyper-methylated sites for FUMA qvalue < 0.01
# Ordered by chromosome and then by methylation difference
sig.hypo <- diffM.df %>%
  filter(P < 0.01 & M < 0) %>%
  arrange(CHR, M)

sig.hyper <- diffM.df %>%
  filter(P < 0.01 & M > 0) %>%
  arrange(CHR, -M)

#write.table(sig.hyper, file="output/3_Annotation/sig_hyper_M20.tsv", row.names=F, quote=F, sep="\t")
#write.table(sig.hypo, file="output/3_Annotation/sig_hypo_M20.tsv", row.names=F, quote=F, sep="\t")
#write.table(sig.hyper, file="output/3_Annotation/sig_hyper.tsv", row.names=F, quote=F, sep="\t")
#write.table(sig.hypo, file="output/3_Annotation/sig_hypo.tsv", row.names=F, quote=F, sep="\t")

diffM.gr <- as(diffM.df,"GRanges")


# export all hypo and hyper methylated sites ** add rsid
getMethylDiff(diffM, difference = 0, qvalue=0.01, type="hypo") %>%
  as.data.frame() %>%
  write_tsv(path="output/3_Annotation/sig_hypo.tsv")

getMethylDiff(diffM, difference = 0, qvalue=0.01, type="hyper") %>%
  as.data.frame() %>%
  write_tsv(path="output/3_Annotation/sig_hyper.tsv")

```
