<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Annat Haber" />


<title>RNAseq Differential Expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MTHFR C667T epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS Pilot
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_Pilot_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_Pilot_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_Pilot_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_Pilot_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNAseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RNAseq_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RNAseq_2_Differential_Expression.html">Differential Expression</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/MTHFR_C667T">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;annat.haber@jax.org&gt;">
    <span class="fas fa-paper-plane"></span>
     
  </a>
</li>
<li>
  <a href="https://www.dropbox.com/home/MTHFR">
    <span class="fab fa-dropbox"></span>
     
  </a>
</li>
<li>
  <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">
    <span class="fas fa-project-diagram"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RNAseq Differential Expression</h1>
<h4 class="author">Annat Haber</h4>
<h4 class="date">5/1/2021</h4>

</div>


<pre class="r"><code>knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = TRUE)</code></pre>
<pre class="r"><code>suppressPackageStartupMessages({
  library(synapser)
  library(DESeq2)
  library(tidyverse)
 })</code></pre>
<div id="input" class="section level2">
<h2>Input</h2>
<ol style="list-style-type: decimal">
<li>Read sample metadata and raw counts.<br />
</li>
<li>Switch genotypes for two samples based on <a href="RNAseq_1_Processing_QC">the QC</a>:<br />
34158 from TT to CT<br />
34163 from CT to TT<br />
</li>
<li>Sort everything by sample ID.</li>
</ol>
<pre class="r"><code># read sample info
meta &lt;- readRDS(&quot;output/RNAseq/1_Processing_QC/samplesheet_meta.RDS&quot;)
meta$genotype[meta$sample_id==&quot;34158&quot;] &lt;- &quot;CT&quot;
meta$genotype[meta$sample_id==&quot;34163&quot;] &lt;- &quot;TT&quot;
meta &lt;- meta %&gt;%
  arrange(genotype, sex, age)

# read raw counts
ctsfile &lt;- &quot;output/RNAseq/1_Processing_QC/nf_rnaseq_output/star_rsem/rsem.merged.gene_counts.tsv&quot;
counts &lt;- read_tsv(ctsfile) %&gt;%
  dplyr::select(-&quot;transcript_id(s)&quot;) %&gt;%
  column_to_rownames(&quot;gene_id&quot;) %&gt;%
  round() %&gt;%
  as.matrix()

# replace pipeline sample id with original sample id
counts &lt;- counts[,meta$samplesheet]
colnames(counts) &lt;- meta$sample_id</code></pre>
</div>
<div id="sample-exploration" class="section level2">
<h2>Sample exploration</h2>
<p>Using variance stabilizing transformation for visualization. There is great separation between male and female. In females there is separation of age, but not in males. There is no separation at all between genotypes.</p>
<pre class="r"><code># Modifying plotPCA function to include PC3
plotPCA = function(object, ntop=500, intgroup)
{
# calculate the variance for each gene
  rv &lt;- rowVars(assay(object))
  # select the ntop genes by variance
  select &lt;- order(rv, decreasing=TRUE)[seq_len(min(ntop, length(rv)))]
  # perform a PCA on the data in assay(x) for the selected genes
  pca &lt;- prcomp(t(assay(object)[select,]))
  # the contribution to the total variance for each component
  percentVar &lt;- pca$sdev^2 / sum( pca$sdev^2 )
  
  intgroup.df &lt;- as.data.frame(colData(object)[, intgroup, drop=FALSE])
  # add the intgroup factors together to create a new grouping factor
  group &lt;- if (length(intgroup) &gt; 1) {
    factor(apply( intgroup.df, 1, paste, collapse=&quot;:&quot;))
  } else {
    colData(object)[[intgroup]]
  }
  # assemble the data for the plot
  d &lt;- data.frame(PC1=pca$x[,1], PC2=pca$x[,2], PC3=pca$x[,3], group=group, intgroup.df, name=colnames(object))
  attr(d, &quot;percentVar&quot;) &lt;- percentVar[1:3]
  return(d)
}</code></pre>
<pre class="r"><code>coldata &lt;- meta %&gt;%
  dplyr::select(sample_id, genotype, sex, age) %&gt;%
  mutate(age=as.numeric(age)) %&gt;%
  column_to_rownames(&quot;sample_id&quot;)

dds &lt;- DESeqDataSetFromMatrix(counts, coldata, design = ~ genotype)
dds &lt;- dds[rowSums(counts(dds)) &gt; 10,]
dds$genotype &lt;- relevel(dds$genotype, ref = &quot;CC&quot;)
vsd &lt;- vst(dds, blind=FALSE)

pcaData &lt;- plotPCA(vsd, intgroup=c(&quot;genotype&quot;, &quot;sex&quot;, &quot;age&quot;))
pcaData$age &lt;- factor(pcaData$age)
percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
ggplot(pcaData, aes(x=PC1,y=PC2)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(1.5,3)) +
  xlab(paste0(&quot;PC1: &quot;,percentVar[1],&quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;,percentVar[2],&quot;% variance&quot;)) + 
  coord_fixed()
ggplot(pcaData, aes(x=PC1,y=PC3)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(1.5,3)) +
  xlab(paste0(&quot;PC1: &quot;,percentVar[1],&quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;,percentVar[3],&quot;% variance&quot;)) + 
  coord_fixed()

rm(dds)</code></pre>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/sample_exp-1.png" width="50%" style="background-color: #9ecff7; padding:1px; display: inline-block;" /><img src="RNAseq_2_Differential_Expression_files/figure-html/sample_exp-2.png" width="50%" style="background-color: #9ecff7; padding:1px; display: inline-block;" /></p>
</div>
<div id="differential-expression" class="section level2 tabset tabset-fade tabset-pills">
<h2>Differential expression</h2>
<div id="variant-vs-wt" class="section level3 tabset tabset-fade tabset-pills">
<h3>Variant vs WT</h3>
<p>For each combination of sex (M,F) and age (6,18) we compare CT to CC and TT to CC.<br />
Genes that have less than 10x reads total are filtered out within each comparison.<br />
Genes with p-adj &lt; 0.1 are in blue.</p>
<pre class="r"><code>grp &lt;- meta %&gt;%
  filter(genotype!=&quot;CC&quot;) %&gt;%
  distinct(genotype, sex, age)

de_resL &lt;- list()
sig.genes_gt &lt;- c()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt &lt;- grp$genotype[i]
  sx &lt;- grp$sex[i]
  ag &lt;- grp$age[i]
  gr &lt;- paste(gt, sx, ag, sep=&quot;_&quot;)
  
  # get sample_id and grouping factor (genotype) for this comparison
  coldata &lt;- meta %&gt;%
    filter(genotype %in% c(&quot;CC&quot;,gt) &amp; sex == sx &amp; age == ag) %&gt;%
    dplyr::select(genotype, sample_id) %&gt;%
    column_to_rownames(&quot;sample_id&quot;)

  cts &lt;- counts[, rownames(coldata)]

  # DESeq object and analysis
  dds &lt;- DESeqDataSetFromMatrix(cts, coldata, design = ~ genotype)
  dds &lt;- dds[rowSums(counts(dds)) &gt; 10,]
  dds$genotype &lt;- relevel(dds$genotype, ref = &quot;CC&quot;)
  dds &lt;- DESeq(dds) 
 
  # extract DEseq results
  res &lt;- results(dds)
  res &lt;- res[order(res),]
  de_resL[[gr]] &lt;- res
  
  # get genes significant at padj &lt; 0.1
  sig.g &lt;- as.data.frame(res[which(res$padj&lt;0.1),])
  if (nrow(sig.g) &gt; 0) {
    sig.g &lt;- cbind(comparison=gr, sig.g) %&gt;%
      arrange(desc(padj)) %&gt;%
      rownames_to_column(var=&#39;ensembl_gene_id&#39;)
    # add gene symbol
    ensembl &lt;- biomaRt::useMart(&quot;ensembl&quot;, dataset=&quot;mmusculus_gene_ensembl&quot;)
    mgi_symbol &lt;- biomaRt::getBM(attributes=c(&#39;ensembl_gene_id&#39;,&#39;external_gene_name&#39;),
             filters = &#39;ensembl_gene_id&#39;,
             values = sig.g$ensembl_gene_id,
             mart = ensembl) %&gt;%
      dplyr::rename(&quot;mgi_symbol&quot;=&quot;external_gene_name&quot;)
    sig.g &lt;- left_join(sig.g, mgi_symbol, by=&#39;ensembl_gene_id&#39;)
    sig.genes_gt &lt;- rbind(sig.genes_gt, sig.g)
  }
  
  rm(gt, sx, ag, coldata, cts, dds, res, sig.g)
}</code></pre>
<pre class="r"><code>for(gr in names(de_resL)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  res &lt;- de_resL[[gr]]
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;CT|TT&quot;, &quot;CC&quot;)), &quot;\n&quot;)
  cat(plotMA(res), &quot;\n&quot;)
  #cat(&quot;\n&quot;, &quot;significant genes:\n&quot;, rownames(res)[which(res$padj&lt;0.1)])
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="ct_f_18" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CC_F_18 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-1.png" width="672" /></p>
</div>
<div id="ct_f_6" class="section level4">
<h4>CT_F_6</h4>
<p>Comparing CT_F_6 to CC_F_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-2.png" width="672" /></p>
</div>
<div id="ct_m_18" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CC_M_18 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-3.png" width="672" /></p>
</div>
<div id="ct_m_6" class="section level4">
<h4>CT_M_6</h4>
<p>Comparing CT_M_6 to CC_M_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-4.png" width="672" /></p>
</div>
<div id="tt_f_18" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to CC_F_18 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-5.png" width="672" /></p>
</div>
<div id="tt_f_6" class="section level4">
<h4>TT_F_6</h4>
<p>Comparing TT_F_6 to CC_F_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-6.png" width="672" /></p>
</div>
<div id="tt_m_18" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to CC_M_18 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-7.png" width="672" /></p>
</div>
<div id="tt_m_6" class="section level4">
<h4>TT_M_6</h4>
<p>Comparing TT_M_6 to CC_M_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_gt-8.png" width="672" /></p>
<pre class="r"><code>de_resL_gt &lt;- de_resL
rm(de_resL)</code></pre>
</div>
</div>
<div id="old-vs-young" class="section level3 tabset tabset-fade tabset-pills">
<h3>Old vs Young</h3>
<p>For each combination of sex (M,F) and genotype (CC, CT, TT) we compare old to young samples (18 vs 6 months).<br />
Genes that have less than 10x reads total are filtered out within each comparison.<br />
Genes with p-adj &lt; 0.1 are in blue.</p>
<pre class="r"><code>grp &lt;- meta %&gt;%
  filter(age!=&quot;6&quot;) %&gt;%
  distinct(genotype, sex, age)

de_resL &lt;- list()
sig.genes_age &lt;- c()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt &lt;- grp$genotype[i]
  sx &lt;- grp$sex[i]
  ag &lt;- grp$age[i]
  gr &lt;- paste(gt, sx, ag, sep=&quot;_&quot;)
  
  # get sample_id and grouping factor (age) for this comparison
  coldata &lt;- meta %&gt;%
    filter(sex == sx &amp; genotype == gt) %&gt;%
    dplyr::select(age, sample_id) %&gt;%
    column_to_rownames(&quot;sample_id&quot;)

  cts &lt;- counts[, rownames(coldata)]

  # DESeq object and analysis
  dds &lt;- DESeqDataSetFromMatrix(cts, coldata, design = ~ age)
  dds &lt;- dds[rowSums(counts(dds)) &gt; 10,]
  dds$age &lt;- relevel(dds$age, ref = &quot;6&quot;)
  dds &lt;- DESeq(dds) # differential expression analysis
 
  # extract DEseq results
  res &lt;- results(dds)
  res &lt;- res[order(res),]
  de_resL[[gr]] &lt;- res
  
  # get genes significant at padj &lt; 0.1
  sig.g &lt;- as.data.frame(res[which(res$padj&lt;0.1),])
  if (nrow(sig.g) &gt; 0) {
    sig.g &lt;- cbind(comparison=gr, sig.g) %&gt;%
      arrange(desc(padj)) %&gt;%
      rownames_to_column(var=&#39;ensembl_gene_id&#39;)
    # add gene symbol
    ensembl &lt;- biomaRt::useMart(&quot;ensembl&quot;, dataset=&quot;mmusculus_gene_ensembl&quot;)
    mgi_symbol &lt;- biomaRt::getBM(attributes=c(&#39;ensembl_gene_id&#39;,&#39;external_gene_name&#39;),
             filters = &#39;ensembl_gene_id&#39;,
             values = sig.g$ensembl_gene_id,
             mart = ensembl) %&gt;%
      dplyr::rename(&quot;mgi_symbol&quot;=&quot;external_gene_name&quot;)
    sig.g &lt;- left_join(sig.g, mgi_symbol, by=&#39;ensembl_gene_id&#39;)
    sig.genes_age &lt;- rbind(sig.genes_age, sig.g)
  }

  rm(gt, sx, ag, coldata, cts, dds, res, sig.g)
}</code></pre>
<pre class="r"><code>for(gr in names(de_resL)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  res &lt;- de_resL[[gr]]
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;18&quot;, &quot;6&quot;)), &quot;\n&quot;)
  cat(plotMA(res), &quot;\n&quot;)
  #cat(&quot;\n&quot;, &quot;significant genes:\n&quot;, rownames(res)[which(res$padj&lt;0.1)])
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="cc_f_18" class="section level4">
<h4>CC_F_18</h4>
<p>Comparing CC_F_18 to CC_F_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_age-1.png" width="672" /></p>
</div>
<div id="cc_m_18" class="section level4">
<h4>CC_M_18</h4>
<p>Comparing CC_M_18 to CC_M_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_age-2.png" width="672" /></p>
</div>
<div id="ct_f_18-1" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CT_F_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_age-3.png" width="672" /></p>
</div>
<div id="ct_m_18-1" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CT_M_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_age-4.png" width="672" /></p>
</div>
<div id="tt_f_18-1" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to TT_F_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_age-5.png" width="672" /></p>
</div>
<div id="tt_m_18-1" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to TT_M_6 <img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res_age-6.png" width="672" /></p>
<pre class="r"><code>de_resL_age &lt;- de_resL
rm(de_resL)</code></pre>
</div>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<pre class="r"><code>write_csv(sig.genes_gt,
            file=&quot;output/RNAseq/2_Differential_Expression/sig_genes_padj01_variantVSwt.csv&quot;)

write_csv(sig.genes_age, 
            file=&quot;output/RNAseq/2_Differential_Expression/sig_genes_padj01_18vs6.csv&quot;)

saveRDS(de_resL_gt, file=&quot;output/RNAseq/2_Differential_Expression/sig.genes_gt.RDS&quot;)
saveRDS(de_resL_age, file=&quot;output/RNAseq/2_Differential_Expression/sig.genes_age.RDS&quot;)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
