<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Annat Haber" />


<title>RNAseq Differential Expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MTHFR C667T epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS Pilot
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_Pilot_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_Pilot_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_Pilot_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_Pilot_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNAseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RNAseq_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RNAseq_2_Differential_Expression.html">Differential Expression</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/MTHFR_C667T">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;annat.haber@jax.org&gt;">
    <span class="fas fa-paper-plane"></span>
     
  </a>
</li>
<li>
  <a href="https://www.dropbox.com/home/MTHFR">
    <span class="fab fa-dropbox"></span>
     
  </a>
</li>
<li>
  <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">
    <span class="fas fa-project-diagram"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RNAseq Differential Expression</h1>
<h4 class="author">Annat Haber</h4>
<h4 class="date">5/1/2021</h4>

</div>


<pre class="r"><code>knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = TRUE)</code></pre>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(synapser)
  library(DESeq2)
  library(&quot;pheatmap&quot;)
  })</code></pre>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Read sample metadata and raw counts.<br />
Sort everything by sample ID.<br />
Filter out genes that have less than 10x reads total.</p>
<pre class="r"><code># read sample info
meta &lt;- readRDS(&quot;output/RNAseq/1_Processing_QC/samplesheet_meta.RDS&quot;)

# read raw counts
ctsfile &lt;- &quot;output/RNAseq/1_Processing_QC/nf_rnaseq_output/star_rsem/rsem.merged.gene_counts.tsv&quot;
counts &lt;- read_tsv(ctsfile) %&gt;%
  dplyr::select(-&quot;transcript_id(s)&quot;) %&gt;%
  column_to_rownames(&quot;gene_id&quot;) %&gt;%
  round() %&gt;%
  as.matrix()

# replace pipeline sample id with original sample id
counts &lt;- counts[,meta$samplesheet]
colnames(counts) &lt;- meta$sample_id</code></pre>
</div>
<div id="deseq-objects-and-results" class="section level2">
<h2>DESeq objects and results</h2>
<p>For each combination of sex (M,F) and age (6,18) we compare CT to CC and TT to CC.</p>
<pre class="r"><code>grp &lt;- meta %&gt;%
  filter(genotype!=&quot;CC&quot;) %&gt;%
  distinct(genotype, sex, age)

ddsL &lt;- de_resL &lt;- list()

for (i in 1:nrow(grp)) {
  gt &lt;- grp$genotype[i]
  sx &lt;- grp$sex[i]
  ag &lt;- grp$age[i]
  gr &lt;- paste(gt, sx, ag, sep=&quot;_&quot;)
  
  coldata &lt;- meta %&gt;%
    filter(genotype %in% c(&quot;CC&quot;,gt) &amp; sex == sx &amp; age == ag) %&gt;%
    select(genotype, sample_id) %&gt;%
    column_to_rownames(&quot;sample_id&quot;)

  cts &lt;- counts[, rownames(coldata)]

  dds &lt;- DESeqDataSetFromMatrix(cts, coldata, design = ~ genotype)
  dds &lt;- dds[rowSums(counts(dds)) &gt; 10,]
  dds$genotype &lt;- relevel(dds$genotype, ref = &quot;CC&quot;)
 
  ddsL[[gr]] &lt;- dds &lt;- DESeq(dds) # differential expression
 
  res &lt;- results(dds)
  res &lt;- res[order(res),]
  de_resL[[gr]] &lt;- res
  
  rm(gt, sx, ag, coldata, cts, dds, res)
}</code></pre>
</div>
<div id="sample-exploration" class="section level2">
<h2>Sample exploration</h2>
<p>Using variance stabilizing transformation for visualization. There is great separation between male and female. In females there is separation of age, but not in males. There is no separation at all between genotypes.</p>
<pre class="r"><code>coldata &lt;- meta %&gt;%
  select(sample_id, genotype, sex, age) %&gt;%
  mutate(age=as.numeric(age)) %&gt;%
  column_to_rownames(&quot;sample_id&quot;)

dds &lt;- DESeqDataSetFromMatrix(counts, coldata, design = ~ genotype)
dds &lt;- dds[rowSums(counts(dds)) &gt; 10,]
dds$genotype &lt;- relevel(dds$genotype, ref = &quot;CC&quot;)
vsd &lt;- vst(dds, blind=FALSE)

pcaData &lt;- plotPCA(vsd, intgroup=c(&quot;genotype&quot;, &quot;sex&quot;, &quot;age&quot;), returnData=TRUE)
pcaData$age &lt;- factor(pcaData$age)
percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
ggplot(pcaData, aes(x=PC1,y=PC2)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(3,1.5)) +
  xlab(paste0(&quot;PC1: &quot;,percentVar[1],&quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;,percentVar[2],&quot;% variance&quot;)) + 
  coord_fixed()</code></pre>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/sample_exp-1.png" width="672" /></p>
</div>
<div id="differential-expression" class="section level2 tabset tabset-fade tabset-pills">
<h2>Differential expression</h2>
<p>Genes with p-adj &lt; 0.1 are in blue.</p>
<pre class="r"><code>for(gr in names(de_resL)){
  cat(&quot;### &quot;, gr ,&quot;\n&quot;)
  res &lt;- de_resL[[gr]]
  #cat(summary(res))
  cat(plotMA(res), &quot;\n&quot;)
  cat(&quot;\n&quot;, &quot;significant genes:\n&quot;, rownames(res[which(res$padj&lt;0.1),]))
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="ct_f_18" class="section level3">
<h3>CT_F_18</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-1.png" width="672" /></p>
<p>significant genes:</p>
</div>
<div id="ct_m_18" class="section level3">
<h3>CT_M_18</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-2.png" width="672" /></p>
<p>significant genes:</p>
</div>
<div id="ct_f_6" class="section level3">
<h3>CT_F_6</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-3.png" width="672" /></p>
<p>significant genes: ENSMUSG00000007207 ENSMUSG00000025417 ENSMUSG00000021559 ENSMUSG00000054808 ENSMUSG00000030329 ENSMUSG00000009013 ENSMUSG00000029436 ENSMUSG00000041708 ENSMUSG00000040640 ENSMUSG00000032773 ENSMUSG00000102543 ENSMUSG00000007950 ENSMUSG00000059173 ENSMUSG00000028514 ENSMUSG00000050821 ENSMUSG00000029581 ENSMUSG00000048644 ENSMUSG00000021466 ENSMUSG00000052387 ENSMUSG00000028152 ENSMUSG00000053550 ENSMUSG00000039671 ENSMUSG00000027221 ENSMUSG00000063229 ENSMUSG00000045994 ENSMUSG00000039126 ENSMUSG00000070498 ENSMUSG00000032470 ENSMUSG00000026259 ENSMUSG00000039943 ENSMUSG00000034818 ENSMUSG00000021062 ENSMUSG00000021373 ENSMUSG00000025372 ENSMUSG00000037428 ENSMUSG00000004936 ENSMUSG00000070047 ENSMUSG00000020589 ENSMUSG00000021775 ENSMUSG00000022099 ENSMUSG00000026090 ENSMUSG00000032368 ENSMUSG00000067879 ENSMUSG00000024411 ENSMUSG00000004187 ENSMUSG00000031543 ENSMUSG00000061689 ENSMUSG00000065954 ENSMUSG00000040125 ENSMUSG00000028782 ENSMUSG00000061911 ENSMUSG00000032936 ENSMUSG00000026452 ENSMUSG00000030653 ENSMUSG00000024403 ENSMUSG00000022010 ENSMUSG00000009569 ENSMUSG00000015829 ENSMUSG00000032966 ENSMUSG00000028832 ENSMUSG00000029657 ENSMUSG00000003363 ENSMUSG00000055430 ENSMUSG00000059456 ENSMUSG00000060961 ENSMUSG00000026204 ENSMUSG00000072214 ENSMUSG00000054728 ENSMUSG00000062296 ENSMUSG00000061751 ENSMUSG00000029120 ENSMUSG00000053025 ENSMUSG00000027347 ENSMUSG00000031778 ENSMUSG00000041444 ENSMUSG00000019302 ENSMUSG00000056486 ENSMUSG00000041773 ENSMUSG00000046093 ENSMUSG00000022421 ENSMUSG00000070570 ENSMUSG00000002265 ENSMUSG00000030209 ENSMUSG00000030695 ENSMUSG00000024617 ENSMUSG00000101111</p>
</div>
<div id="ct_m_6" class="section level3">
<h3>CT_M_6</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-4.png" width="672" /></p>
<p>significant genes:</p>
</div>
<div id="tt_f_18" class="section level3">
<h3>TT_F_18</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-5.png" width="672" /></p>
<p>significant genes:</p>
</div>
<div id="tt_m_18" class="section level3">
<h3>TT_M_18</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-6.png" width="672" /></p>
<p>significant genes: ENSMUSG00000105703</p>
</div>
<div id="tt_f_6" class="section level3">
<h3>TT_F_6</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-7.png" width="672" /></p>
<p>significant genes: ENSMUSG00000093445</p>
</div>
<div id="tt_m_6" class="section level3">
<h3>TT_M_6</h3>
<p><img src="RNAseq_2_Differential_Expression_files/figure-html/deseq_res-8.png" width="672" /></p>
<p>significant genes: ENSMUSG00000000889</p>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>Uploading to synapse</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
