<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Annat Haber" />

<meta name="date" content="2021-05-07" />

<title>RRBS Pilot Methylation by Position</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MTHFR C667T epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS Pilot
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_Pilot_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_Pilot_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_Pilot_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_Pilot_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNAseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RNAseq_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RNAseq_2_Differential_Expression.html">Differential Expression</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/MTHFR_C667T">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;annat.haber@jax.org&gt;">
    <span class="fas fa-paper-plane"></span>
     
  </a>
</li>
<li>
  <a href="https://www.dropbox.com/home/MTHFR">
    <span class="fab fa-dropbox"></span>
     
  </a>
</li>
<li>
  <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">
    <span class="fas fa-project-diagram"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RRBS Pilot Methylation by Position</h1>
<h4 class="author">Annat Haber</h4>
<h4 class="date">2021-05-07</h4>

</div>


<pre class="r"><code>knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(message=FALSE, warning=FALSE, cache = FALSE, cache.path = &quot;output/RRBS_Pilot/4_Methylation_Position&quot;, eval=FALSE)</code></pre>
<pre class="r"><code>rm(list=ls())

suppressPackageStartupMessages({
  library(tidyverse)
  library(GenomicRanges)
  library(biomaRt)
  })

source(&quot;scripts/AnnaTyler/center.on.feature.R&quot;)
source(&quot;scripts/AnnaTyler/plot.centered.vals.R&quot;)
source(&quot;scripts/AnnaTyler/plot.poly.xy.R&quot;)</code></pre>
<p>Load data</p>
<pre class="r"><code># a matrix of methyl percent per site by sample; only sites that are shared by all samples
meth &lt;- readRDS(&quot;output/2_Analysis/precentMethylation.RDS&quot;)

# metadata
meta &lt;- read_csv(&quot;output/2_Analysis/sampleInfo.globalmeth.csv&quot;)

# reference annotations from http://genome.ucsc.edu/cgi-bin/hgTables
refseq &lt;- read_tsv(&quot;data/mm10.refseq.genes.txt&quot;)</code></pre>
<p>Gene info for all known mouse genes<br />
rows are genes; columns are external_gene_name, start_position, end_position, strand</p>
<pre class="r"><code>mart &lt;- useMart(biomart=&quot;ensembl&quot;, dataset=&quot;mmusculus_gene_ensembl&quot;)
gene.info &lt;- getBM(attributes = c(&quot;external_gene_name&quot;, &quot;chromosome_name&quot;, &quot;start_position&quot;, &quot;end_position&quot;, &quot;strand&quot;), filters = c(&quot;external_gene_name&quot;), values = unique(refseq$name2), mart = mart)
del &lt;- which(is.na(as.numeric(gene.info$chromosome_name)))
gene.info &lt;- gene.info[-del,]</code></pre>
<p>Map snps to genes, including downstream and upstream buffer zones</p>
<pre class="r"><code>buffer &lt;- 2000
ext.start &lt;- gene.info$start_position - buffer
ext.end &lt;- gene.info$end_position + buffer

snps &lt;- rownames(meth) %&gt;%
  strsplit(split=&quot;:&quot;) %&gt;% unlist() %&gt;% 
  matrix(nr=nrow(meth), nc=2, byrow = TRUE) %&gt;%
  apply(2, as.numeric)
snps &lt;- snps[!is.na(snps[,1]),] # removing X and Y

snps.genes &lt;- c()
for (i in 1:nrow(snps)) {
  chr &lt;- snps[i,1]
  pos &lt;- snps[i,2]
  genes.i &lt;- which(gene.info$chromosome_name == chr &amp;
                 ext.start &lt; pos &amp;
                 ext.end &gt; pos)
  if (length(genes.i)&gt;0) {
    snps.genes &lt;- rbind(snps.genes, 
                      cbind(pos, gene=gene.info$external_gene_name[genes.i]))
  }
}</code></pre>
<p>Get methyl values by gene, averaged across replicates<br />
Coordinates are normalized within gene<br />
One vector for CC and one for TT for each gene</p>
<pre class="r"><code>gene.methyl.norm &lt;- list()
for (tr in c(&quot;CC&quot;, &quot;TT&quot;)) {
  for (g in unique(snps.genes[,&quot;gene&quot;])) {
    chr &lt;- gene.info$chromosome_name[gene.info$external_gene_name==g]
    pos.g &lt;-  paste(chr, snps.genes[snps.genes[,&quot;gene&quot;]==g, &quot;pos&quot;], sep=&quot;:&quot;)
    if (length(pos.g)&gt;1) {
      # extracting and averaging methyl values across replicates
      # mg &lt;- meth[pos.g, meta$genotype==tr &amp; meta$age==&quot;13&quot;] %&gt;% rowMeans()
      mg &lt;- meth[pos.g, meta$genotype==tr] %&gt;% rowMeans()
     names(mg) &lt;- matrix(unlist(strsplit(names(mg), split=&quot;:&quot;)), nc=2, byrow=TRUE)[,2]
      # centering on TSS and normalizing by gene length
      gene.methyl.norm[[tr]][[g]] &lt;- center.on.feature(
                                      gene.name = g, # mgi_symbol
                                      gene.info = gene.info, 
                                      vals = mg, # methyl vals of one gene with coordinates as names
                                      feature = &quot;full&quot;)
    }
  }
}</code></pre>
<p>Aligning and ploting</p>
<pre class="r"><code># aligning the centered normallized methylation values for CC averaged across genes
cc.aligned.means &lt;- plot.centered.vals(val.list = gene.methyl.norm[[&quot;CC&quot;]], 
plot.label = &quot;CC Genes&quot;, ylim = c(0, 500), min.representation = 1, seq.by = 0.01, merge.by = 0,
plot.individual = FALSE, return.means=TRUE, min.upstream = -2000, max.downstream = 2000)

# centered normallized methylation values for TT averaged across genes
tt.aligned.means &lt;- plot.centered.vals(val.list = gene.methyl.norm[[&quot;TT&quot;]], 
plot.label = &quot;TT Genes&quot;, ylim = c(0, 500), min.representation =1, seq.by = 0.01, merge.by = 0,
plot.individual = FALSE, return.means=TRUE, min.upstream = -2000, max.downstream = 2000)

# position values
cc.pos &lt;- as.numeric(names(cc.aligned.means))
tt.pos &lt;- as.numeric(names(tt.aligned.means)) 

line.col &lt;- c(rgb(216/256, 179/256, 101/256), rgb(1/256, 133/256, 113/256, alpha = 0.5))

# plot
#quartz(width = 9, height = 5)
plot.new()
plot.window(ylim = c(0, 100), xlim = c(-2, 2))
abline(v = c(0, 1), lwd = 1, col = &quot;black&quot;)
text(x = 0.5, y = 100, labels = &quot;gene body&quot;)
arrows(x0 = 0, x1 = 1, y0 = 95)
points(cc.pos, cc.aligned.means, type = &quot;l&quot;, col = line.col[1])
points(tt.pos, tt.aligned.means, type = &quot;l&quot;, col = line.col[2])
mtext(&quot;Relative Position&quot;, side = 1, line = 2)
mtext(&quot;Percent Methylation&quot;, side = 2, line = 2)
legend(&quot;topright&quot;, col = line.col, lty = 1, 
legend = c(&quot;CC&quot;, &quot;TT&quot;), lwd = 3)
axis(1);axis(2)


plot(cc.aligned.means, tt.aligned.means)
segments(0,0,100,100, col=&quot;red&quot;)

### BINNED
# aligning the centered normallized methylation values for CC averaged across genes
cc.aligned.means &lt;- plot.centered.vals(val.list = gene.methyl.norm[[&quot;CC&quot;]], 
plot.label = &quot;CC Genes&quot;, ylim = c(0, 500), min.representation = 1, seq.by = 0.05, merge.by = 0,
plot.individual = FALSE, return.means=TRUE, min.upstream = -2000, max.downstream = 2000)

# centered normallized methylation values for TT averaged across genes
tt.aligned.means &lt;- plot.centered.vals(val.list = gene.methyl.norm[[&quot;TT&quot;]], 
plot.label = &quot;TT Genes&quot;, ylim = c(0, 500), min.representation =1, seq.by = 0.05, merge.by = 0,
plot.individual = FALSE, return.means=TRUE, min.upstream = -2000, max.downstream = 2000)

# position values
cc.pos &lt;- as.numeric(names(cc.aligned.means))
tt.pos &lt;- as.numeric(names(tt.aligned.means)) 

line.col &lt;- c(rgb(216/256, 179/256, 101/256), rgb(1/256, 133/256, 113/256, alpha = 0.5))

# plot
#quartz(width = 9, height = 5)
plot.new()
plot.window(ylim = c(0, 100), xlim = c(-2, 2))
abline(v = c(0, 1), lwd = 1, col = &quot;black&quot;)
text(x = 0.5, y = 100, labels = &quot;gene body&quot;)
arrows(x0 = 0, x1 = 1, y0 = 95)
points(cc.pos, cc.aligned.means, type = &quot;l&quot;, col = line.col[1])
points(tt.pos, tt.aligned.means, type = &quot;l&quot;, col = line.col[2])
mtext(&quot;Relative Position&quot;, side = 1, line = 2)
mtext(&quot;Percent Methylation&quot;, side = 2, line = 2)
legend(&quot;topright&quot;, col = line.col, lty = 1, 
legend = c(&quot;CC&quot;, &quot;TT&quot;), lwd = 3)
axis(1);axis(2)</code></pre>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
