<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Annat Haber" />

<meta name="date" content="2021-06-04" />

<title>RRBS Differential Methylation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
#rmd-source-code {
  display: none;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MTHFR C667T epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS Pilot
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_Pilot_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_Pilot_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_Pilot_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_Pilot_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNAseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RNAseq_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RNAseq_2_Differential_Expression.html">Differential Expression</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation_gt.html">Differential Methylation; pooled sex and age</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation_gtsxag.html">Differential Methylation; separate sex and age</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/MTHFR_C667T">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;annat.haber@jax.org&gt;">
    <span class="fas fa-paper-plane"></span>
     
  </a>
</li>
<li>
  <a href="https://www.dropbox.com/home/MTHFR">
    <span class="fab fa-dropbox"></span>
     
  </a>
</li>
<li>
  <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">
    <span class="fas fa-project-diagram"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RRBS Differential Methylation</h1>
<h3 class="subtitle">Separated by genotype, age, and sex</h3>
<h4 class="author">Annat Haber</h4>
<h4 class="date">2021-06-04</h4>

</div>


<pre class="r"><code>knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(cache = TRUE, 
                      #cache.path = &quot;output/RRBS/3_Differential_Methylation&quot;,
                      message=FALSE, warning=FALSE)</code></pre>
<pre class="r"><code>rm(list=ls())

suppressPackageStartupMessages({
  library(methylKit)
  library(synapser)
  library(tidyverse)
  library(GenomicRanges)
  library(genomation)
  library(ChIPpeakAnno)
  library(clusterProfiler)
  library(GOsummaries)
  library(org.Mm.eg.db)
  library(biomaRt)
  library(vcd)
})</code></pre>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Read sample metadata and methylation calls.<br />
Genotypes were already switched for two samples (34158 from TT to CT; 34163 from CT to TT)</p>
<pre class="r"><code>united_meth &lt;- readRDS(file=&quot;output/RRBS/1_Processing_QC/united_meth.RDS&quot;)
meta &lt;- read_csv(file=&quot;output/RRBS/2_Exploratory_Analysis/RRBS_globalmeth_hmc.csv&quot;,
                 col_types = cols(.default = &quot;c&quot;))</code></pre>
<pre class="r"><code>synLogin() 

united_meth &lt;- readRDS(synGet(&quot;syn25834047&quot;)$path)
meta &lt;- read_csv(synGet(&quot;syn25834048&quot;)$path,
                 col_types = cols(.default = &quot;c&quot;))</code></pre>
</div>
<div id="differentially-methylated-cpg-sites" class="section level2 tabset tabset-fade tabset-pills">
<h2>differentially Methylated CpG sites</h2>
<p>Finding differentially methylated CpG sites separately for age groups and sex.<br />
Mapping to known genes by finding overlap between the snp’s and mm10 reference.<br />
The reference annotation was downloaded directly from <a href="http://genome.ucsc.edu/cgi-bin/hgTables" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables</a></p>
<pre class="r"><code># read reference and convert to GRanges
refseq &lt;- read_tsv(&quot;data/mm10.refseq.genes.txt&quot;) 

ref.gr &lt;- GRanges(
        seqnames = substr(refseq$chrom, 4,5),
        ranges=IRanges(start = refseq$txStart, end = refseq$txEnd),
        mgi_name=refseq$name2
        )
names(ref.gr) &lt;- refseq$name
rm(refseq)</code></pre>
<div id="variant-vs-wt" class="section level3 tabset tabset-fade tabset-pills">
<h3>Variant vs WT</h3>
<p>For each combination of sex (M,F) and age (6,18) we compare CT to CC and TT to CC. Top genes are identified as significant at qvalue &lt; 0.01 and methylation difference &gt;10</p>
<pre class="r"><code>grp &lt;- meta %&gt;%
  filter(genotype!=&quot;CC&quot;) %&gt;%
  distinct(genotype, sex, age)

diffML_gt &lt;- sig.snpL_gt &lt;- list()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt &lt;- grp$genotype[i]
  sx &lt;- grp$sex[i]
  ag &lt;- grp$age[i]
  gr &lt;- paste(gt, sx, ag, sep=&quot;_&quot;)
  
  # get sample_id and grouping factor (genotype) for this comparison
  cov &lt;- meta %&gt;%
    filter(genotype %in% c(&quot;CC&quot;,gt) &amp; sex == sx &amp; age == ag) %&gt;%
    mutate(genotype = ifelse(genotype==gt, 1, 0)) %&gt;%
    dplyr::select(sample_id, genotype, sex, age)
  
  # DiffMeth object and analysis
  diffM &lt;- reorganize(united_meth, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$genotype) %&gt;%
            calculateDiffMeth()
  
  # extract significant sites and convert to GRanges
  snp.gr &lt;- getMethylDiff(diffM, difference = 10, qvalue=0.01, type=&quot;all&quot;) %&gt;%
    as(&quot;GRanges&quot;)
  names(snp.gr) &lt;- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=&quot;:&quot;)

  # Find overlap with reference
  overlap &lt;- findOverlaps(snp.gr, ref.gr)
  hits &lt;- ref.gr$mgi_name[slot(overlap, &quot;to&quot;)]
  names(hits) &lt;- names(snp.gr)[slot(overlap, &quot;from&quot;)]
  hits &lt;- hits[!duplicated(hits)]
  snpDMgenes &lt;- snp.gr[names(hits),]
  mcols(snpDMgenes)$mgi_name &lt;- hits
  
  snpDMgenes &lt;- data.frame(snpDMgenes) %&gt;%
    dplyr::select(chromosome=seqnames, everything()) %&gt;%
    arrange(chromosome, meth.diff)
  
  diffML_gt[[gr]] &lt;- diffM
  sig.snpL_gt[[gr]] &lt;- snpDMgenes
  
  rm(gt, sx, ag, cov, diffM, snp.gr, overlap, hits, snpDMgenes)
}</code></pre>
<pre class="r"><code>for(gr in names(diffML_gt)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  diffM &lt;- diffML_gt[[gr]]
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;CT|TT&quot;, &quot;CC&quot;)), &quot;\n&quot;)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="ct_f_18" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CC_F_18 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-1.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-2.png" width="672" /></p>
</div>
<div id="ct_m_18" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CC_M_18 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-3.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-4.png" width="672" /></p>
</div>
<div id="ct_f_6" class="section level4">
<h4>CT_F_6</h4>
<p>Comparing CT_F_6 to CC_F_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-5.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-6.png" width="672" /></p>
</div>
<div id="ct_m_6" class="section level4">
<h4>CT_M_6</h4>
<p>Comparing CT_M_6 to CC_M_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-7.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-8.png" width="672" /></p>
</div>
<div id="tt_f_18" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to CC_F_18 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-9.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-10.png" width="672" /></p>
</div>
<div id="tt_m_18" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to CC_M_18 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-11.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-12.png" width="672" /></p>
</div>
<div id="tt_f_6" class="section level4">
<h4>TT_F_6</h4>
<p>Comparing TT_F_6 to CC_F_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-13.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-14.png" width="672" /></p>
</div>
<div id="tt_m_6" class="section level4">
<h4>TT_M_6</h4>
<p>Comparing TT_M_6 to CC_M_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-15.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_gt-16.png" width="672" /></p>
<pre class="r"><code>rm(diffM)</code></pre>
</div>
</div>
<div id="old-vs-young" class="section level3 tabset tabset-fade tabset-pills">
<h3>Old vs Young</h3>
<p>For each combination of sex (M,F) and genotype (CC, CT, TT) we compare old to young samples (18 vs 6 months).<br />
Top genes are identified as significant at qvalue &lt; 0.01 and methylation difference &gt;10</p>
<pre class="r"><code>grp &lt;- meta %&gt;%
  filter(age==&quot;18&quot;) %&gt;%
  distinct(genotype, sex, age)

diffML_age &lt;- sig.snpL_age &lt;- list()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt &lt;- grp$genotype[i]
  sx &lt;- grp$sex[i]
  ag &lt;- grp$age[i]
  gr &lt;- paste(gt, sx, ag, sep=&quot;_&quot;)
  
  # get sample_id and grouping factor (genotype) for this comparison
  cov &lt;- meta %&gt;%
    filter(sex == sx &amp; genotype == gt) %&gt;%
    mutate(age = ifelse(age==&quot;18&quot;, 1, 0)) %&gt;%
    dplyr::select(sample_id, genotype, sex, age)
  
  # DiffMeth object and analysis
  diffM &lt;- reorganize(united_meth, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$age) %&gt;%
            calculateDiffMeth()
  
  # extract significant sites and convert to GRanges
  snp.gr &lt;- getMethylDiff(diffM, difference = 10, qvalue=0.01, type=&quot;all&quot;) %&gt;%
    as(&quot;GRanges&quot;)
  names(snp.gr) &lt;- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=&quot;:&quot;)

  # Find overlap with reference
  overlap &lt;- findOverlaps(snp.gr, ref.gr)
  hits &lt;- ref.gr$mgi_name[slot(overlap, &quot;to&quot;)]
  names(hits) &lt;- names(snp.gr)[slot(overlap, &quot;from&quot;)]
  hits &lt;- hits[!duplicated(hits)]
  snpDMgenes &lt;- snp.gr[names(hits),]
  mcols(snpDMgenes)$mgi_name &lt;- hits
  
  snpDMgenes &lt;- data.frame(snpDMgenes) %&gt;%
    dplyr::select(chromosome=seqnames, everything()) %&gt;%
    arrange(chromosome, meth.diff)
  
  diffML_age[[gr]] &lt;- diffM
  sig.snpL_age[[gr]] &lt;- snpDMgenes
  
  rm(gt, sx, ag, cov, diffM, snp.gr, overlap, hits, snpDMgenes)
}</code></pre>
<pre class="r"><code>for(gr in names(diffML_age)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  diffM &lt;- diffML_age[[gr]]
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;18&quot;, &quot;6&quot;)), &quot;\n&quot;)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="cc_f_18" class="section level4">
<h4>CC_F_18</h4>
<p>Comparing CC_F_18 to CC_F_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-1.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-2.png" width="672" /></p>
</div>
<div id="cc_m_18" class="section level4">
<h4>CC_M_18</h4>
<p>Comparing CC_M_18 to CC_M_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-3.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-4.png" width="672" /></p>
</div>
<div id="ct_f_18-1" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CT_F_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-5.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-6.png" width="672" /></p>
</div>
<div id="ct_m_18-1" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CT_M_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-7.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-8.png" width="672" /></p>
</div>
<div id="tt_f_18-1" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to TT_F_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-9.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-10.png" width="672" /></p>
</div>
<div id="tt_m_18-1" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to TT_M_6 <img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-11.png" width="672" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/diffM_plot_age-12.png" width="672" /></p>
<pre class="r"><code>rm(diffM)</code></pre>
</div>
</div>
</div>
<div id="gene-parts-annotation" class="section level2 tabset tabset-fade tabset-pills">
<h2>Gene parts annotation</h2>
<p>Gene parts distribution for differentially-methylated sites, defines with qvalue &lt; 0.01 and methylation difference &gt;10</p>
<pre class="r"><code>gene.obj=readTranscriptFeatures(&quot;data/mm10.refseq.genes.bed&quot;)
cpg.obj=readFeatureFlank(&quot;data/mm10.refseq.genes.bed&quot;,
                           feature.flank.name=c(&quot;CpGi&quot;,&quot;shores&quot;))</code></pre>
<div id="variant-vs-wt-1" class="section level3 tabset tabset-fade tabset-pills">
<h3>Variant vs WT</h3>
<pre class="r"><code>for(gr in names(diffML_gt)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  diffM &lt;- diffML_gt[[gr]]
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;CT|TT&quot;, &quot;CC&quot;)), &quot;\n&quot;, &quot;\n&quot;)
  # hypermethylated sites with difference &gt; 10%
  diffM10.hyper &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hyper&quot;) %&gt;%
    as(&quot;GRanges&quot;)
  annotateWithGeneParts(diffM10.hyper,  gene.obj) %&gt;%
     plotTargetAnnotation(precedence=TRUE,
                       main=&quot;Hyper-methylated gene parts; &gt; 10% diff.&quot;)
  
  annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;) %&gt;%
    plotTargetAnnotation(col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
                       main=&quot;Hyper-methylated flank annotation; &gt; 10% diff.&quot;)


  # hypomethylated sites with difference &gt; 10%
  diffM10.hypo &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hypo&quot;) %&gt;%
    as(&quot;GRanges&quot;)
  annotateWithGeneParts(diffM10.hypo,  gene.obj) %&gt;%
     plotTargetAnnotation(precedence=TRUE,
                       main=&quot;Hypo-methylated gene parts; &gt; 10% diff.&quot;)
  
  annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;) %&gt;%
    plotTargetAnnotation(col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
                       main=&quot;Hypo-methylated flank annotation; &gt; 10% diff.&quot;)
  
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="ct_f_18-2" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CC_F_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-1.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-2.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-3.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-4.png" width="50%" /></p>
</div>
<div id="ct_m_18-2" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CC_M_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-5.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-6.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-7.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-8.png" width="50%" /></p>
</div>
<div id="ct_f_6-1" class="section level4">
<h4>CT_F_6</h4>
<p>Comparing CT_F_6 to CC_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-9.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-10.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-11.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-12.png" width="50%" /></p>
</div>
<div id="ct_m_6-1" class="section level4">
<h4>CT_M_6</h4>
<p>Comparing CT_M_6 to CC_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-13.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-14.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-15.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-16.png" width="50%" /></p>
</div>
<div id="tt_f_18-2" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to CC_F_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-17.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-18.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-19.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-20.png" width="50%" /></p>
</div>
<div id="tt_m_18-2" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to CC_M_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-21.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-22.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-23.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-24.png" width="50%" /></p>
</div>
<div id="tt_f_6-1" class="section level4">
<h4>TT_F_6</h4>
<p>Comparing TT_F_6 to CC_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-25.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-26.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-27.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-28.png" width="50%" /></p>
</div>
<div id="tt_m_6-1" class="section level4">
<h4>TT_M_6</h4>
<p>Comparing TT_M_6 to CC_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-29.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-30.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-31.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_gt-32.png" width="50%" /></p>
</div>
</div>
<div id="old-vs-young-1" class="section level3 tabset tabset-fade tabset-pills">
<h3>Old vs Young</h3>
<pre class="r"><code>for(gr in names(diffML_age)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  diffM &lt;- diffML_age[[gr]]
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;18&quot;, &quot;6&quot;)), &quot;\n&quot;, &quot;\n&quot;)
  # hypermethylated sites with difference &gt; 10%
  diffM10.hyper &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hyper&quot;) %&gt;%
    as(&quot;GRanges&quot;)
  annotateWithGeneParts(diffM10.hyper,  gene.obj) %&gt;%
     plotTargetAnnotation(precedence=TRUE,
                       main=&quot;Hyper-methylated gene parts; &gt; 10% diff.&quot;)
  
  annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;) %&gt;%
    plotTargetAnnotation(col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
                       main=&quot;Hyper-methylated flank annotation; &gt; 10% diff.&quot;)


  # hypomethylated sites with difference &gt; 10%
  diffM10.hypo &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hypo&quot;) %&gt;%
    as(&quot;GRanges&quot;)
  annotateWithGeneParts(diffM10.hypo,  gene.obj) %&gt;%
     plotTargetAnnotation(precedence=TRUE,
                       main=&quot;Hypo-methylated gene parts; &gt; 10% diff.&quot;)
  
  annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;) %&gt;%
    plotTargetAnnotation(col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
                       main=&quot;Hypo-methylated flank annotation; &gt; 10% diff.&quot;)
  
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="cc_f_18-1" class="section level4">
<h4>CC_F_18</h4>
<p>Comparing CC_F_18 to CC_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-1.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-2.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-3.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-4.png" width="50%" /></p>
</div>
<div id="cc_m_18-1" class="section level4">
<h4>CC_M_18</h4>
<p>Comparing CC_M_18 to CC_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-5.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-6.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-7.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-8.png" width="50%" /></p>
</div>
<div id="ct_f_18-3" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CT_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-9.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-10.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-11.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-12.png" width="50%" /></p>
</div>
<div id="ct_m_18-3" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CT_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-13.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-14.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-15.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-16.png" width="50%" /></p>
</div>
<div id="tt_f_18-3" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to TT_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-17.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-18.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-19.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-20.png" width="50%" /></p>
</div>
<div id="tt_m_18-3" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to TT_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-21.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-22.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-23.png" width="50%" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/gene_parts_age-24.png" width="50%" /></p>
</div>
</div>
</div>
<div id="kegg-annotation" class="section level2 tabset tabset-fade tabset-pills">
<h2>KEGG Annotation</h2>
<div id="variant-vs-wt-2" class="section level3 tabset tabset-fade tabset-pills">
<h3>Variant vs WT</h3>
<pre class="r"><code>keggL_gt &lt;- list()

for (gr in names(sig.snpL_gt)) {
  
  ## hyper-methylated genes
  sig.gs &lt;- sig.snpL_gt[[gr]] %&gt;%
    dplyr::filter(meth.diff &gt; 0) %&gt;% arrange(meth.diff) %&gt;%
    pull(mgi_name)
  # entrez id
  sig.gs.ez &lt;- bitr(sig.gs, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Mm.eg.db&quot;)$ENTREZID
  # kegg annotation
  anno &lt;- enrichKEGG(gene = sig.gs.ez,
                organism = &quot;mmu&quot;,
                keyType = &quot;ncbi-geneid&quot;,
                pvalueCutoff = 0.5,
                pAdjustMethod = &quot;BH&quot;,
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno &lt;- setReadable(anno, OrgDb = org.Mm.eg.db, keyType=&quot;ENTREZID&quot;)
  keggL_gt[[gr]][[&quot;hyper&quot;]] &lt;- anno

  ## hypo-methylated genes
  sig.gs &lt;- sig.snpL_gt[[gr]] %&gt;%
    dplyr::filter(meth.diff &lt; 0) %&gt;% arrange(meth.diff) %&gt;%
    pull(mgi_name)
  # entrez id
  sig.gs.ez &lt;- bitr(sig.gs, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Mm.eg.db&quot;)$ENTREZID
  # kegg annotation
  anno &lt;- enrichKEGG(gene = sig.gs.ez,
                organism = &quot;mmu&quot;,
                keyType = &quot;ncbi-geneid&quot;,
                pvalueCutoff = 0.5,
                pAdjustMethod = &quot;BH&quot;,
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno &lt;- setReadable(anno, OrgDb = org.Mm.eg.db, keyType=&quot;ENTREZID&quot;)
  keggL_gt[[gr]][[&quot;hypo&quot;]] &lt;- anno
}

rm(sig.gs, sig.gs.ez, anno)</code></pre>
<pre class="r"><code>for(gr in names(keggL_gt)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;CT|TT&quot;, &quot;CC&quot;)), &quot;\n&quot;, &quot;\n&quot;)

  anno &lt;- keggL_gt[[gr]][[&quot;hyper&quot;]]
  print(barplot(anno, showCategory=20, title=paste0(&quot;KEGG pathways for hyper-methylated sites&quot;)))
  anno &lt;- as.data.frame(anno@result) %&gt;%
    mutate(meth_dir=&quot;hyper&quot;)
  write_csv(anno, file=paste0(&quot;output/RRBS/3_Differential_Methylation/kegg_gt_&quot;,gr,&quot;.csv&quot;))

  anno &lt;- keggL_gt[[gr]][[&quot;hypo&quot;]]
  print(barplot(anno, showCategory=20, title=paste0(&quot;KEGG pathways for hypo-methylated sites&quot;)))
  anno &lt;- as.data.frame(anno@result) %&gt;%
    mutate(meth_dir=&quot;hypo&quot;)
  write_csv(anno, file=paste0(&quot;output/RRBS/3_Differential_Methylation/kegg_gt_&quot;,gr,&quot;.csv&quot;), append = TRUE)
  
  cat(&#39;\n\n&#39;)

}</code></pre>
<div id="ct_f_18-4" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CC_F_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-1.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-2.png" width="960" /></p>
</div>
<div id="ct_m_18-4" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CC_M_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-3.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-4.png" width="960" /></p>
</div>
<div id="ct_f_6-2" class="section level4">
<h4>CT_F_6</h4>
<p>Comparing CT_F_6 to CC_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-5.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-6.png" width="960" /></p>
</div>
<div id="ct_m_6-2" class="section level4">
<h4>CT_M_6</h4>
<p>Comparing CT_M_6 to CC_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-7.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-8.png" width="960" /></p>
</div>
<div id="tt_f_18-4" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to CC_F_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-9.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-10.png" width="960" /></p>
</div>
<div id="tt_m_18-4" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to CC_M_18</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-11.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-12.png" width="960" /></p>
</div>
<div id="tt_f_6-2" class="section level4">
<h4>TT_F_6</h4>
<p>Comparing TT_F_6 to CC_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-13.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-14.png" width="960" /></p>
</div>
<div id="tt_m_6-2" class="section level4">
<h4>TT_M_6</h4>
<p>Comparing TT_M_6 to CC_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-15.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_gt-16.png" width="960" /></p>
<pre class="r"><code>rm(anno)</code></pre>
</div>
</div>
<div id="old-vs-young-2" class="section level3 tabset tabset-fade tabset-pills">
<h3>Old vs Young</h3>
<pre class="r"><code>keggL_age &lt;- list()

for (gr in names(sig.snpL_age)) {
  
  ## hyper-methylated genes
  sig.gs &lt;- sig.snpL_age[[gr]] %&gt;%
    dplyr::filter(meth.diff &gt; 0) %&gt;% arrange(meth.diff) %&gt;%
    pull(mgi_name)
  # entrez id
  sig.gs.ez &lt;- bitr(sig.gs, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Mm.eg.db&quot;)$ENTREZID
  # kegg annotation
  anno &lt;- enrichKEGG(gene = sig.gs.ez,
                organism = &quot;mmu&quot;,
                keyType = &quot;ncbi-geneid&quot;,
                pvalueCutoff = 0.5,
                pAdjustMethod = &quot;BH&quot;,
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno &lt;- setReadable(anno, OrgDb = org.Mm.eg.db, keyType=&quot;ENTREZID&quot;)
  keggL_age[[gr]][[&quot;hyper&quot;]] &lt;- anno

  ## hypo-methylated genes
  sig.gs &lt;- sig.snpL_age[[gr]] %&gt;%
    dplyr::filter(meth.diff &lt; 0) %&gt;% arrange(meth.diff) %&gt;%
    pull(mgi_name)
  # entrez id
  sig.gs.ez &lt;- bitr(sig.gs, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Mm.eg.db&quot;)$ENTREZID
  # kegg annotation
  anno &lt;- enrichKEGG(gene = sig.gs.ez,
                organism = &quot;mmu&quot;,
                keyType = &quot;ncbi-geneid&quot;,
                pvalueCutoff = 0.5,
                pAdjustMethod = &quot;BH&quot;,
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno &lt;- setReadable(anno, OrgDb = org.Mm.eg.db, keyType=&quot;ENTREZID&quot;)
  keggL_age[[gr]][[&quot;hypo&quot;]] &lt;- anno
}

rm(sig.gs, sig.gs.ez, anno)</code></pre>
<pre class="r"><code>for(gr in names(keggL_age)){
  cat(&quot;#### &quot;, gr ,&quot;\n&quot;)
  cat(paste0(&quot;Comparing &quot;, gr, &quot; to &quot;, str_replace(gr, &quot;18&quot;, &quot;6&quot;)), &quot;\n&quot;, &quot;\n&quot;)

  anno &lt;- keggL_age[[gr]][[&quot;hyper&quot;]]
  print(barplot(anno, showCategory=20, title=paste0(&quot;KEGG pathways for hyper-methylated sites&quot;)))
  anno &lt;- as.data.frame(anno@result) %&gt;%
    mutate(meth_dir=&quot;hyper&quot;)
  write_csv(anno, file=paste0(&quot;output/RRBS/3_Differential_Methylation/kegg_age_&quot;,gr,&quot;.csv&quot;))

  anno &lt;- keggL_age[[gr]][[&quot;hypo&quot;]]
  print(barplot(anno, showCategory=20, title=paste0(&quot;KEGG pathways for hypo-methylated sites&quot;)))
  anno &lt;- as.data.frame(anno@result) %&gt;%
    mutate(meth_dir=&quot;hypo&quot;)
  write_csv(anno, file=paste0(&quot;output/RRBS/3_Differential_Methylation/kegg_age_&quot;,gr,&quot;.csv&quot;), append = TRUE)
  
  cat(&#39;\n\n&#39;)

}</code></pre>
<div id="cc_f_18-2" class="section level4">
<h4>CC_F_18</h4>
<p>Comparing CC_F_18 to CC_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-1.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-2.png" width="960" /></p>
</div>
<div id="cc_m_18-2" class="section level4">
<h4>CC_M_18</h4>
<p>Comparing CC_M_18 to CC_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-3.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-4.png" width="960" /></p>
</div>
<div id="ct_f_18-5" class="section level4">
<h4>CT_F_18</h4>
<p>Comparing CT_F_18 to CT_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-5.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-6.png" width="960" /></p>
</div>
<div id="ct_m_18-5" class="section level4">
<h4>CT_M_18</h4>
<p>Comparing CT_M_18 to CT_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-7.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-8.png" width="960" /></p>
</div>
<div id="tt_f_18-5" class="section level4">
<h4>TT_F_18</h4>
<p>Comparing TT_F_18 to TT_F_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-9.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-10.png" width="960" /></p>
</div>
<div id="tt_m_18-5" class="section level4">
<h4>TT_M_18</h4>
<p>Comparing TT_M_18 to TT_M_6</p>
<p><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-11.png" width="960" /><img src="RRBS_3_Differential_Methylation_gtsxag_files/figure-html/kegg_plot_age-12.png" width="960" /></p>
<pre class="r"><code>rm(anno)</code></pre>
</div>
</div>
</div>
<div id="cell-specificity" class="section level2 tabset tabset-fade tabset-pills">
<h2>Cell-specificity</h2>
<p>Cell type markers in the brain are taken from McKenzie et al. (Supplemental File 1, <a href="DOI:10.1038/s41598-018-27293-5" class="uri">DOI:10.1038/s41598-018-27293-5</a>). Cell specificity for a given gene was determined in this study as minimum log fold change when pairwise comparing the expression of the gene in the given cell type with the expression of the gene in all other cell types. Cell type for that gene was then determined as the cell type with the highest specificity. The top 1000 genes were included in the final dataset as markers for each cell type.</p>
<p>Here I’m using cell specificity score to look for overlap between cell-specific genes and hypo/hyper methylated genes. I use chi-square to test for biased representations of any cell type genes in either the hypo- or hyper-methylated genes, and a plot of chi-square deviations from expectation for visualization.</p>
<div id="variant-vs-wt-3" class="section level3 tabset tabset-fade tabset-pills">
<h3>Variant vs WT</h3>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>Saving lists of significantly hyper/hypo methylated genes to project folder</p>
<pre class="r"><code># variant vs wt
write_csv(as.tibble(t(c(colnames(sig.snpL_gt[[1]]), &quot;comparison&quot;))), 
            file=&quot;output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv&quot;,
          col_names = F)

for (gr in names(sig.snpL_gt)) {
  sig.snpL_gt[[gr]] %&gt;%
    dplyr::mutate(comparison=gr) %&gt;%
    write_csv(file=&quot;output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv&quot;, append=TRUE)
  }

# age 18 vs 6
write_csv(as.tibble(t(c(colnames(sig.snpL_age[[1]]), &quot;comparison&quot;))), 
            file=&quot;output/RRBS/3_Differential_Methylation/sig_dmg_age.csv&quot;,
          col_names = F)

for (gr in names(sig.snpL_age)) {
  sig.snpL_age[[gr]] %&gt;%
    dplyr::mutate(comparison=gr) %&gt;%
    write_csv(file=&quot;output/RRBS/3_Differential_Methylation/sig_dmg_age.csv&quot;, append=TRUE)
  }</code></pre>
<p>Uploading lists of significant genes to [synapse]((<a href="https://www.synapse.org/#!Synapse:syn25174486" class="uri">https://www.synapse.org/#!Synapse:syn25174486</a>)</p>
<pre class="r"><code>synLogin() 
## variant vs wt
file &lt;- File(&quot;output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv&quot;,
             contentType = &quot;text/plain&quot;,
             description = &quot;lists of significantly hyper/hypo methylated genes; variant vs wt&quot;,
             parent = &quot;syn25174486&quot;)

provenance &lt;- Activity(used = c(&quot;syn25834048&quot;, &quot;syn25834047&quot;),
                       executed = &quot;https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_3_Differential_Methylation_gtsxag.Rmd&quot;)

file &lt;- synStore(file, activity=provenance)

# 18 vs 6 months
file &lt;- File(&quot;output/RRBS/3_Differential_Methylation/sig_dmg_age.csv&quot;,
             contentType = &quot;text/plain&quot;,
             description = &quot;lists of significantly hyper/hypo methylated genes; 18 vs 6 months&quot;,
             parent = &quot;syn25174486&quot;)

provenance &lt;- Activity(used = c(&quot;syn25834048&quot;, &quot;syn25834047&quot;),
                       executed = &quot;https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_3_Differential_Methylation_gtsxag.Rmd&quot;)

file &lt;- synStore(file, activity=provenance)</code></pre>
</div>

<div id="rmd-source-code">---
title: "RRBS Differential Methylation"
subtitle: "Separated by genotype, age, and sex"
author: "Annat Haber"
date: '`r Sys.Date()`'
output: 
  html_document:
    toc: true
    toc_depth: 3
    code_folding: hide
---

```{r setup, warning=FALSE, message=FALSE}
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(cache = TRUE, 
                      #cache.path = "output/RRBS/3_Differential_Methylation",
                      message=FALSE, warning=FALSE)
```

```{r libraries}
rm(list=ls())

suppressPackageStartupMessages({
  library(methylKit)
  library(synapser)
  library(tidyverse)
  library(GenomicRanges)
  library(genomation)
  library(ChIPpeakAnno)
  library(clusterProfiler)
  library(GOsummaries)
  library(org.Mm.eg.db)
  library(biomaRt)
  library(vcd)
})

```

## Input
Read sample metadata and methylation calls.  
Genotypes were already switched for two samples (34158 from TT to CT; 34163 from CT to TT)
```{r input_project}
united_meth <- readRDS(file="output/RRBS/1_Processing_QC/united_meth.RDS")
meta <- read_csv(file="output/RRBS/2_Exploratory_Analysis/RRBS_globalmeth_hmc.csv",
                 col_types = cols(.default = "c"))
```

```{r input_synapse, eval=FALSE}
synLogin() 

united_meth <- readRDS(synGet("syn25834047")$path)
meta <- read_csv(synGet("syn25834048")$path,
                 col_types = cols(.default = "c"))
```

## differentially Methylated CpG sites  {.tabset .tabset-fade .tabset-pills}
Finding differentially methylated CpG sites separately for age groups and sex.  
Mapping to known genes by finding overlap between the snp's and mm10 reference.  
The reference annotation was downloaded directly from http://genome.ucsc.edu/cgi-bin/hgTables 
```{r refseq_gr}
# read reference and convert to GRanges
refseq <- read_tsv("data/mm10.refseq.genes.txt") 

ref.gr <- GRanges(
        seqnames = substr(refseq$chrom, 4,5),
        ranges=IRanges(start = refseq$txStart, end = refseq$txEnd),
        mgi_name=refseq$name2
        )
names(ref.gr) <- refseq$name
rm(refseq)
```

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
For each combination of sex (M,F) and age (6,18) we compare CT to CC and TT to CC.
Top genes are identified as significant at qvalue < 0.01 and methylation difference >10
```{r diffM_obj_gt}
grp <- meta %>%
  filter(genotype!="CC") %>%
  distinct(genotype, sex, age)

diffML_gt <- sig.snpL_gt <- list()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt <- grp$genotype[i]
  sx <- grp$sex[i]
  ag <- grp$age[i]
  gr <- paste(gt, sx, ag, sep="_")
  
  # get sample_id and grouping factor (genotype) for this comparison
  cov <- meta %>%
    filter(genotype %in% c("CC",gt) & sex == sx & age == ag) %>%
    mutate(genotype = ifelse(genotype==gt, 1, 0)) %>%
    dplyr::select(sample_id, genotype, sex, age)
  
  # DiffMeth object and analysis
  diffM <- reorganize(united_meth, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$genotype) %>%
            calculateDiffMeth()
  
  # extract significant sites and convert to GRanges
  snp.gr <- getMethylDiff(diffM, difference = 10, qvalue=0.01, type="all") %>%
    as("GRanges")
  names(snp.gr) <- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=":")

  # Find overlap with reference
  overlap <- findOverlaps(snp.gr, ref.gr)
  hits <- ref.gr$mgi_name[slot(overlap, "to")]
  names(hits) <- names(snp.gr)[slot(overlap, "from")]
  hits <- hits[!duplicated(hits)]
  snpDMgenes <- snp.gr[names(hits),]
  mcols(snpDMgenes)$mgi_name <- hits
  
  snpDMgenes <- data.frame(snpDMgenes) %>%
    dplyr::select(chromosome=seqnames, everything()) %>%
    arrange(chromosome, meth.diff)
  
  diffML_gt[[gr]] <- diffM
  sig.snpL_gt[[gr]] <- snpDMgenes
  
  rm(gt, sx, ag, cov, diffM, snp.gr, overlap, hits, snpDMgenes)
}
```

```{r, diffM_plot_gt, message=FALSE, warning=FALSE, fig.height = 6, results='asis'}
for(gr in names(diffML_gt)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_gt[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n")
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)
  cat('\n\n')
}

rm(diffM)
```

### Old vs Young {.tabset .tabset-fade .tabset-pills}
For each combination of sex (M,F) and genotype (CC, CT, TT) we compare old to young samples (18 vs 6 months).  
Top genes are identified as significant at qvalue < 0.01 and methylation difference >10
```{r diffM_obj_age}
grp <- meta %>%
  filter(age=="18") %>%
  distinct(genotype, sex, age)

diffML_age <- sig.snpL_age <- list()

for (i in 1:nrow(grp)) {
  #print(i)
  # get genotype, sex, and age for this comparison
  gt <- grp$genotype[i]
  sx <- grp$sex[i]
  ag <- grp$age[i]
  gr <- paste(gt, sx, ag, sep="_")
  
  # get sample_id and grouping factor (genotype) for this comparison
  cov <- meta %>%
    filter(sex == sx & genotype == gt) %>%
    mutate(age = ifelse(age=="18", 1, 0)) %>%
    dplyr::select(sample_id, genotype, sex, age)
  
  # DiffMeth object and analysis
  diffM <- reorganize(united_meth, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$age) %>%
            calculateDiffMeth()
  
  # extract significant sites and convert to GRanges
  snp.gr <- getMethylDiff(diffM, difference = 10, qvalue=0.01, type="all") %>%
    as("GRanges")
  names(snp.gr) <- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=":")

  # Find overlap with reference
  overlap <- findOverlaps(snp.gr, ref.gr)
  hits <- ref.gr$mgi_name[slot(overlap, "to")]
  names(hits) <- names(snp.gr)[slot(overlap, "from")]
  hits <- hits[!duplicated(hits)]
  snpDMgenes <- snp.gr[names(hits),]
  mcols(snpDMgenes)$mgi_name <- hits
  
  snpDMgenes <- data.frame(snpDMgenes) %>%
    dplyr::select(chromosome=seqnames, everything()) %>%
    arrange(chromosome, meth.diff)
  
  diffML_age[[gr]] <- diffM
  sig.snpL_age[[gr]] <- snpDMgenes
  
  rm(gt, sx, ag, cov, diffM, snp.gr, overlap, hits, snpDMgenes)
}
```

```{r, diffM_plot_age, message=FALSE, warning=FALSE, fig.height = 6, results='asis'}
for(gr in names(diffML_age)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_age[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "18", "6")), "\n")
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)
  diffMethPerChr(diffM, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)
  cat('\n\n')
}

rm(diffM)
```

## Gene parts annotation {.tabset .tabset-fade .tabset-pills}
Gene parts distribution for differentially-methylated sites, defines with qvalue < 0.01 and methylation difference >10
```{r ref_anno}
gene.obj=readTranscriptFeatures("data/mm10.refseq.genes.bed")
cpg.obj=readFeatureFlank("data/mm10.refseq.genes.bed",
                           feature.flank.name=c("CpGi","shores"))
```

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
```{r, gene_parts_gt, message=FALSE, warning=FALSE, results='asis', out.width=c('50%', '50%')}
for(gr in names(diffML_gt)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_gt[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n", "\n")
  # hypermethylated sites with difference > 10%
  diffM10.hyper <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hyper") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hyper,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hyper-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hyper-methylated flank annotation; > 10% diff.")


  # hypomethylated sites with difference > 10%
  diffM10.hypo <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hypo") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hypo,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hypo-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hypo-methylated flank annotation; > 10% diff.")
  
  cat('\n\n')
}

```

### Old vs Young {.tabset .tabset-fade .tabset-pills}
```{r, gene_parts_age, message=FALSE, warning=FALSE, results='asis', out.width=c('50%', '50%')}
for(gr in names(diffML_age)){
  cat("#### ", gr ,"\n")
  diffM <- diffML_age[[gr]]
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "18", "6")), "\n", "\n")
  # hypermethylated sites with difference > 10%
  diffM10.hyper <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hyper") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hyper,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hyper-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hyper-methylated flank annotation; > 10% diff.")


  # hypomethylated sites with difference > 10%
  diffM10.hypo <- getMethylDiff(diffM, difference=10, qvalue=0.01, type="hypo") %>%
    as("GRanges")
  annotateWithGeneParts(diffM10.hypo,  gene.obj) %>%
     plotTargetAnnotation(precedence=TRUE,
                       main="Hypo-methylated gene parts; > 10% diff.")
  
  annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                                    feature.name="CpGi",flank.name="shores") %>%
    plotTargetAnnotation(col=c("green","gray","white"),
                       main="Hypo-methylated flank annotation; > 10% diff.")
  
  cat('\n\n')
}

```

## KEGG Annotation {.tabset .tabset-fade .tabset-pills}

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
```{r, kegg_gt, results="hide"}
keggL_gt <- list()

for (gr in names(sig.snpL_gt)) {
  
  ## hyper-methylated genes
  sig.gs <- sig.snpL_gt[[gr]] %>%
    dplyr::filter(meth.diff > 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_gt[[gr]][["hyper"]] <- anno

  ## hypo-methylated genes
  sig.gs <- sig.snpL_gt[[gr]] %>%
    dplyr::filter(meth.diff < 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_gt[[gr]][["hypo"]] <- anno
}

rm(sig.gs, sig.gs.ez, anno)
```

```{r, kegg_plot_gt, results="asis", fig.width = 10}
for(gr in names(keggL_gt)){
  cat("#### ", gr ,"\n")
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n", "\n")

  anno <- keggL_gt[[gr]][["hyper"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hyper-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hyper")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_gt_",gr,".csv"))

  anno <- keggL_gt[[gr]][["hypo"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hypo-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hypo")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_gt_",gr,".csv"), append = TRUE)
  
  cat('\n\n')

}

rm(anno)
```

### Old vs Young {.tabset .tabset-fade .tabset-pills}
```{r, kegg_age, results="hide"}
keggL_age <- list()

for (gr in names(sig.snpL_age)) {
  
  ## hyper-methylated genes
  sig.gs <- sig.snpL_age[[gr]] %>%
    dplyr::filter(meth.diff > 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_age[[gr]][["hyper"]] <- anno

  ## hypo-methylated genes
  sig.gs <- sig.snpL_age[[gr]] %>%
    dplyr::filter(meth.diff < 0) %>% arrange(meth.diff) %>%
    pull(mgi_name)
  # entrez id
  sig.gs.ez <- bitr(sig.gs, fromType="SYMBOL", toType="ENTREZID", OrgDb="org.Mm.eg.db")$ENTREZID
  # kegg annotation
  anno <- enrichKEGG(gene = sig.gs.ez,
                organism = "mmu",
                keyType = "ncbi-geneid",
                pvalueCutoff = 0.5,
                pAdjustMethod = "BH",
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  anno <- setReadable(anno, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
  keggL_age[[gr]][["hypo"]] <- anno
}

rm(sig.gs, sig.gs.ez, anno)

```

```{r, kegg_plot_age, results="asis", fig.width = 10}
for(gr in names(keggL_age)){
  cat("#### ", gr ,"\n")
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "18", "6")), "\n", "\n")

  anno <- keggL_age[[gr]][["hyper"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hyper-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hyper")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_age_",gr,".csv"))

  anno <- keggL_age[[gr]][["hypo"]]
  print(barplot(anno, showCategory=20, title=paste0("KEGG pathways for hypo-methylated sites")))
  anno <- as.data.frame(anno@result) %>%
    mutate(meth_dir="hypo")
  write_csv(anno, file=paste0("output/RRBS/3_Differential_Methylation/kegg_age_",gr,".csv"), append = TRUE)
  
  cat('\n\n')

}

rm(anno)
```

## Cell-specificity {.tabset .tabset-fade .tabset-pills}
Cell type markers in the brain are taken from McKenzie et al. (Supplemental File 1, DOI:10.1038/s41598-018-27293-5).
Cell specificity for a given gene was determined in this study as minimum log fold change when pairwise comparing the expression of the gene in the given cell type with the expression of the gene in all other cell types. Cell type for that gene was then determined as the cell type with the highest specificity. The top 1000 genes were included in the final dataset as markers for each cell type.

Here I'm using cell specificity score to look for overlap between cell-specific genes and hypo/hyper methylated genes. I use chi-square to test for biased representations of any cell type genes in either the hypo- or hyper-methylated genes, and a plot of chi-square deviations from expectation for visualization.  

```{r cell_spec_ref, eval=FALSE, include=FALSE}
cell.types <- c("Astroctyes", "Endothelial", "Microglia", "Neurons", "Oligodendrocytes")
sets <- c("hypo","hyper")

# read cell-specificity reference
hum.specific <- read.csv("data/McKenzie_2017_Brain_Cell_Specific_Markers_Human_Specificity.csv", header=T, stringsAsFactors=F)
hum.specific <- split(hum.specific, hum.specific$Celltype, drop=TRUE) 
names(hum.specific) <- cell.types

```

### Variant vs WT {.tabset .tabset-fade .tabset-pills}
```{r, cell_spec_gt, results="asis", eval=FALSE, include=FALSE}

for (gr in names(sig.snpL_gt)) {
  cat("#### ", gr ,"\n")
  cat(paste0("Comparing ", gr, " to ", str_replace(gr, "CT|TT", "CC")), "\n", "\n")
  sig.gs <- sig.snpL_gt[[gr]]

# get gene sets
gene.sets <- lapply(sig.gs, function(l){dplyr::select(l, mgi_name)})

gene.sets.hg <- list()
for (s in names(sig.gs)) {
    gene.sets.hg[[s]] = getLDS(attributes = c("mgi_symbol"), filters = "mgi_symbol", 
                              values = sig.gs[[s]]$mgi_name, mart = mouse,
                              attributesL = c("hgnc_symbol"), martL = human, uniqueRows=T)$HGNC.symbol

}

# Number of marker genes for each cell type in each gene set
specific.set <- sapply(sets, function(s) {
  sapply(cell.types, function(ct) length(intersect(hum.specific[[ct]]$gene, gene.sets.hg[[s]])), USE.NAMES=T)
})

print(specific.set)
chisq.test(specific.set)
assoc(t(specific.set), shade=TRUE,labeling_args = list(abbreviate_labs = c(6, 9)))

}
rm(sig.gs, gene.sets, gene.sets.hg, specific.set)

```

## Output
Saving lists of significantly hyper/hypo methylated genes to project folder
```{r output_save}
# variant vs wt
write_csv(as.tibble(t(c(colnames(sig.snpL_gt[[1]]), "comparison"))), 
            file="output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv",
          col_names = F)

for (gr in names(sig.snpL_gt)) {
  sig.snpL_gt[[gr]] %>%
    dplyr::mutate(comparison=gr) %>%
    write_csv(file="output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv", append=TRUE)
  }

# age 18 vs 6
write_csv(as.tibble(t(c(colnames(sig.snpL_age[[1]]), "comparison"))), 
            file="output/RRBS/3_Differential_Methylation/sig_dmg_age.csv",
          col_names = F)

for (gr in names(sig.snpL_age)) {
  sig.snpL_age[[gr]] %>%
    dplyr::mutate(comparison=gr) %>%
    write_csv(file="output/RRBS/3_Differential_Methylation/sig_dmg_age.csv", append=TRUE)
  }

```

Uploading lists of significant genes to [synapse]((https://www.synapse.org/#!Synapse:syn25174486)
```{r output_synapse, results='hide'}
synLogin() 
## variant vs wt
file <- File("output/RRBS/3_Differential_Methylation/sig_dmg_gt.csv",
             contentType = "text/plain",
             description = "lists of significantly hyper/hypo methylated genes; variant vs wt",
             parent = "syn25174486")

provenance <- Activity(used = c("syn25834048", "syn25834047"),
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_3_Differential_Methylation_gtsxag.Rmd")

file <- synStore(file, activity=provenance)

# 18 vs 6 months
file <- File("output/RRBS/3_Differential_Methylation/sig_dmg_age.csv",
             contentType = "text/plain",
             description = "lists of significantly hyper/hypo methylated genes; 18 vs 6 months",
             parent = "syn25174486")

provenance <- Activity(used = c("syn25834048", "syn25834047"),
                       executed = "https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_3_Differential_Methylation_gtsxag.Rmd")

file <- synStore(file, activity=provenance)

```
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("RRBS_3_Differential_Methylation_gtsxag.Rmd");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
