<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Annat Haber" />

<meta name="date" content="2021-05-07" />

<title>RRBS Differential Methylation Annotation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MTHFR C667T epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS Pilot
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_Pilot_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_Pilot_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_Pilot_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_Pilot_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNAseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RNAseq_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RNAseq_2_Differential_Expression.html">Differential Expression</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/MTHFR_C667T">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;annat.haber@jax.org&gt;">
    <span class="fas fa-paper-plane"></span>
     
  </a>
</li>
<li>
  <a href="https://www.dropbox.com/home/MTHFR">
    <span class="fab fa-dropbox"></span>
     
  </a>
</li>
<li>
  <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">
    <span class="fas fa-project-diagram"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RRBS Differential Methylation Annotation</h1>
<h4 class="author">Annat Haber</h4>
<h4 class="date">2021-05-07</h4>

</div>


<pre class="r"><code>knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
knitr::opts_chunk$set(cache = TRUE, 
                      #cache.path = &quot;output/RRBS/3_Differential_Methylation&quot;,
                      message=FALSE, warning=FALSE)</code></pre>
<pre class="r"><code>rm(list=ls())

suppressPackageStartupMessages({
  library(methylKit)
  library(tidyverse)

  library(GenomicRanges)
  library(genomation)
  library(ChIPpeakAnno)
  library(clusterProfiler)
  library(GOsummaries)
  library(org.Mm.eg.db)
  library(biomaRt)
  library(vcd)
})</code></pre>
<div id="input" class="section level2">
<h2>Input</h2>
<pre class="r"><code>united_meth &lt;- readRDS(file=&quot;output/RRBS/1_Processing_QC/united_meth.RDS&quot;)
meta &lt;- read_csv(file=&quot;output/RRBS/2_Exploratory_Analysis/RRBS_meta_globalmeth.csv&quot;,
                 col_types = cols(.default = &quot;c&quot;))</code></pre>
</div>
<div id="differentially-methylated-cpg-sites" class="section level2 tabset tabset-fade tabset-pills">
<h2>differentially Methylated CpG sites</h2>
<p>Finding differentially methylated CpG sites using age as covariate.<br />
Mapping to known genes by finding overlap between the snp’s and mm10 reference.<br />
The reference annotation was downloaded directly from <a href="http://genome.ucsc.edu/cgi-bin/hgTables" class="uri">http://genome.ucsc.edu/cgi-bin/hgTables</a></p>
<pre class="r"><code># read reference and convert to GRanges
refseq &lt;- read_tsv(&quot;data/mm10.refseq.genes.txt&quot;) 

ref.gr &lt;- GRanges(
        seqnames = substr(refseq$chrom, 4,5),
        ranges=IRanges(start = refseq$txStart, end = refseq$txEnd),
        mgi_name=refseq$name2
        )
names(ref.gr) &lt;- refseq$name
rm(refseq)</code></pre>
<div id="tt-vs-cc" class="section level3">
<h3>TT vs CC</h3>
<pre class="r"><code>mm &lt;- united_meth
cov &lt;- meta %&gt;%
  filter(genotype!=&quot;CT&quot;) %&gt;%
  mutate(genotype = ifelse(genotype==&quot;TT&quot;, 1, 0))
  
diffM_TT &lt;- reorganize(mm, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$genotype) %&gt;%
            calculateDiffMeth(covariates = as.data.frame(cov$age))

rm(mm, cov)</code></pre>
<pre class="r"><code>diffM &lt;- diffM_TT
# extract significant sites and convert to GRanges
snp.gr &lt;- getMethylDiff(diffM, difference = 10, qvalue=0.01, type=&quot;all&quot;) %&gt;%
  as(&quot;GRanges&quot;)
names(snp.gr) &lt;- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=&quot;:&quot;)

# Find overlap with reference
overlap &lt;- findOverlaps(snp.gr, ref.gr)
hits &lt;- ref.gr$mgi_name[slot(overlap, &quot;to&quot;)]
names(hits) &lt;- names(snp.gr)[slot(overlap, &quot;from&quot;)]
hits &lt;- hits[!duplicated(hits)]
snpDMgenes &lt;- snp.gr[names(hits),]
mcols(snpDMgenes)$mgi_name &lt;- hits

snpDMgenes &lt;- data.frame(snpDMgenes) %&gt;%
  dplyr::select(chromosome=seqnames, everything()) %&gt;%
  arrange(chromosome, meth.diff)

# separate hypo and hyper methylated gene sets
snpDMgenes_TT &lt;- list(&quot;hypo&quot;= snpDMgenes %&gt;%
                        filter(qvalue &lt; 0.01 &amp; meth.diff &lt; 0) %&gt;%
                        arrange(meth.diff),
                      &quot;hyper&quot;= snpDMgenes %&gt;%
                        filter(qvalue &lt; 0.01 &amp; meth.diff &gt; 0) %&gt;%
                        arrange(-meth.diff)
                      )

rm(diffM, snp.gr, snpDMgenes, hits, overlap)</code></pre>
<pre class="r"><code>diffMethPerChr(diffM_TT, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/diffM_TT_plot-1.png" width="672" /></p>
<pre class="r"><code>diffMethPerChr(diffM_TT, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/diffM_TT_plot-2.png" width="672" /></p>
</div>
<div id="ct-vs-cc" class="section level3">
<h3>CT vs CC</h3>
<pre class="r"><code>mm &lt;- united_meth
cov &lt;- meta %&gt;%
  filter(genotype!=&quot;TT&quot;) %&gt;%
  mutate(genotype = ifelse(genotype==&quot;CT&quot;, 1, 0))
  
diffM_CT &lt;- reorganize(mm, 
                       sample.ids = cov$sample_id, 
                       treatment= cov$genotype) %&gt;%
            calculateDiffMeth(covariates = as.data.frame(cov$age))

rm(mm, cov)</code></pre>
<pre class="r"><code>diffM &lt;- diffM_CT
# extract significant sites and convert to GRanges
snp.gr &lt;- getMethylDiff(diffM, difference = 0, qvalue=0.01, type=&quot;all&quot;) %&gt;%
  as(&quot;GRanges&quot;)
names(snp.gr) &lt;- paste(snp.gr@seqnames@values, snp.gr@ranges@start, sep=&quot;:&quot;)

# Find overlap with reference
overlap &lt;- findOverlaps(snp.gr, ref.gr)
hits &lt;- ref.gr$mgi_name[slot(overlap, &quot;to&quot;)]
names(hits) &lt;- names(snp.gr)[slot(overlap, &quot;from&quot;)]
hits &lt;- hits[!duplicated(hits)]
snpDMgenes &lt;- snp.gr[names(hits),]
mcols(snpDMgenes)$mgi_name &lt;- hits

snpDMgenes &lt;- data.frame(snpDMgenes) %&gt;%
  dplyr::select(chromosome=seqnames, everything()) %&gt;%
  arrange(chromosome, meth.diff)

# separate hypo and hyper methylated gene sets
snpDMgenes_CT &lt;- list(&quot;hypo&quot;= snpDMgenes %&gt;%
                        filter(qvalue &lt; 0.01 &amp; meth.diff &lt; 0) %&gt;%
                        arrange(meth.diff),
                      &quot;hyper&quot;= snpDMgenes %&gt;%
                        filter(qvalue &lt; 0.01 &amp; meth.diff &gt; 0) %&gt;%
                        arrange(-meth.diff)
                      )

rm(diffM, snp.gr, snpDMgenes, hits, overlap)</code></pre>
<pre class="r"><code>diffMethPerChr(diffM_CT, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=10)</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/diffM_CT_plot-1.png" width="672" /></p>
<pre class="r"><code>diffMethPerChr(diffM_CT, plot=TRUE, qvalue.cutoff=0.01, meth.cutoff=0)</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/diffM_CT_plot-2.png" width="672" /></p>
</div>
</div>
<div id="gene-parts-annotation" class="section level2 tabset tabset-fade tabset-pills">
<h2>Gene parts annotation</h2>
<p>Gene parts distribution for site that are differentially-methylated</p>
<div id="tt-vs-cc-1" class="section level3">
<h3>TT vs CC</h3>
<pre class="r"><code>diffM &lt;- diffM_TT

# Get hypermethylated sites with difference &gt; 10% and convert to GRanges
diffM10.hyper &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hyper&quot;) %&gt;%
  as(&quot;GRanges&quot;)

# Get hypomethylated sites with difference &gt; 10% and convert to GRanges
diffM10.hypo &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hypo&quot;) %&gt;%
  as(&quot;GRanges&quot;)


# Get gene parts annotation
gene.obj=readTranscriptFeatures(&quot;data/mm10.refseq.genes.bed&quot;)

hypo10.ann &lt;- annotateWithGeneParts(diffM10.hypo,  gene.obj)

hyper10.ann &lt;- annotateWithGeneParts(diffM10.hyper, gene.obj)

# Get flank annotation
cpg.obj=readFeatureFlank(&quot;data/mm10.refseq.genes.bed&quot;,
                           feature.flank.name=c(&quot;CpGi&quot;,&quot;shores&quot;))

diffCpGann.hypo10 &lt;- annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                         feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;)

diffCpGann.hyper10 &lt;- annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                         feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;)
rm(diffM)</code></pre>
<pre class="r"><code>plotTargetAnnotation(hypo10.ann, precedence=TRUE,
    main=&quot;Hypo-methylated gene parts; &gt; 10% diff.&quot;)

plotTargetAnnotation(hyper10.ann, precedence=TRUE,
    main=&quot;Hyper-methylated gene parts; &gt; 10% diff.&quot;)


plotTargetAnnotation(diffCpGann.hypo10,col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
       main=&quot;Hypo-methylated flank annotation; &gt; 10% diff.&quot;)

plotTargetAnnotation(diffCpGann.hyper10,col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
       main=&quot;Hyper-methylated flank annotation; &gt; 10% diff.&quot;)</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_TT-1.png" width="50%" /><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_TT-2.png" width="50%" /><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_TT-3.png" width="50%" /><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_TT-4.png" width="50%" /></p>
</div>
<div id="ct-vs-cc-1" class="section level3">
<h3>CT vs CC</h3>
<pre class="r"><code>diffM &lt;- diffM_CT

# Get hypermethylated sites with difference &gt; 10% and convert to GRanges
diffM10.hyper &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hyper&quot;) %&gt;%
  as(&quot;GRanges&quot;)

# Get hypomethylated sites with difference &gt; 10% and convert to GRanges
diffM10.hypo &lt;- getMethylDiff(diffM, difference=10, qvalue=0.01, type=&quot;hypo&quot;) %&gt;%
  as(&quot;GRanges&quot;)


# Get gene parts annotation
gene.obj=readTranscriptFeatures(&quot;data/mm10.refseq.genes.bed&quot;)

hypo10.ann &lt;- annotateWithGeneParts(diffM10.hypo,  gene.obj)

hyper10.ann &lt;- annotateWithGeneParts(diffM10.hyper, gene.obj)

# Get flank annotation
cpg.obj=readFeatureFlank(&quot;data/mm10.refseq.genes.bed&quot;,
                           feature.flank.name=c(&quot;CpGi&quot;,&quot;shores&quot;))

diffCpGann.hypo10 &lt;- annotateWithFeatureFlank(diffM10.hypo,
                                    cpg.obj$CpGi,cpg.obj$shores,
                         feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;)

diffCpGann.hyper10 &lt;- annotateWithFeatureFlank(diffM10.hyper,
                                    cpg.obj$CpGi,cpg.obj$shores,
                         feature.name=&quot;CpGi&quot;,flank.name=&quot;shores&quot;)
rm(diffM)</code></pre>
<pre class="r"><code>plotTargetAnnotation(hypo10.ann, precedence=TRUE,
    main=&quot;Hypo-methylated gene parts&quot;)

plotTargetAnnotation(hyper10.ann, precedence=TRUE,
    main=&quot;Hyper-methylated gene parts&quot;)


plotTargetAnnotation(diffCpGann.hypo10,col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
       main=&quot;Hypo-methylated flank annotation&quot;)

plotTargetAnnotation(diffCpGann.hyper10,col=c(&quot;green&quot;,&quot;gray&quot;,&quot;white&quot;),
       main=&quot;Hyper-methylated flank annotation&quot;)</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_CT-1.png" width="50%" /><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_CT-2.png" width="50%" /><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_CT-3.png" width="50%" /><img src="RRBS_3_Differential_Methylation_files/figure-html/gene_parts_plots_CT-4.png" width="50%" /></p>
</div>
</div>
<div id="kegg-annotation" class="section level2 tabset tabset-fade tabset-pills">
<h2>KEGG Annotation</h2>
<div id="tt-vs-cc-2" class="section level3">
<h3>TT vs CC</h3>
<pre class="r"><code>sig.gs &lt;- snpDMgenes_TT

# get entrez id
sig.gs.ez &lt;- list()
for (gs in names(sig.gs)) {
  sig.gs.ez[[gs]] &lt;- bitr(sig.gs[[gs]]$mgi_name, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Mm.eg.db&quot;)$ENTREZID
}

kegg_anno_TT &lt;- list()
for (gs in names(sig.gs.ez)) {
  cat(&quot;\n Annotating:&quot;, gs,&quot;-methylated genes\n&quot;)
  
  anno &lt;- enrichKEGG(gene = sig.gs.ez[[gs]],
                organism = &quot;mmu&quot;,
                keyType = &quot;ncbi-geneid&quot;,
                pvalueCutoff = 0.5,
                pAdjustMethod = &quot;BH&quot;,
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  
  print(barplot(anno, showCategory=20, title=paste0(&quot;KEGG pathways for &quot;, gs, &quot;-methylated sites&quot;)))
  anno &lt;- setReadable(anno, OrgDb = org.Mm.eg.db, keyType=&quot;ENTREZID&quot;)
  anno &lt;- as.data.frame(anno@result)
  write_csv(anno, file=paste0(&quot;output/RRBS/3_Differential_Methylation/kegg_anno_TT_&quot;, gs, &quot;.csv&quot;))

  kegg_anno_TT[[gs]] &lt;- anno
  rm(anno)
  }</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/kegg_annotation_TT-1.png" width="960" /><img src="RRBS_3_Differential_Methylation_files/figure-html/kegg_annotation_TT-2.png" width="960" /></p>
<pre class="r"><code>rm(sig.gs, sig.gs.ez)</code></pre>
</div>
<div id="ct-vs-cc-2" class="section level3">
<h3>CT vs CC</h3>
<pre class="r"><code>sig.gs &lt;- snpDMgenes_CT

# get entrez id
sig.gs.ez &lt;- list()
for (gs in names(sig.gs)) {
  sig.gs.ez[[gs]] &lt;- bitr(sig.gs[[gs]]$mgi_name, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Mm.eg.db&quot;)$ENTREZID
}
kegg_anno_CT &lt;- list()
for (gs in names(sig.gs.ez)) {
  cat(&quot;\n Annotating:&quot;, gs,&quot;-methylated genes\n&quot;)
  
  anno &lt;- enrichKEGG(gene = sig.gs.ez[[gs]],
                organism = &quot;mmu&quot;,
                keyType = &quot;ncbi-geneid&quot;,
                pvalueCutoff = 0.5,
                pAdjustMethod = &quot;BH&quot;,
                minGSSize = 10,
                maxGSSize = 500,
                qvalueCutoff = 0.5,
                use_internal_data = FALSE)
  
  print(barplot(anno, showCategory=20, title=paste0(&quot;KEGG pathways for &quot;, gs, &quot;-methylated sites&quot;)))
  anno &lt;- setReadable(anno, OrgDb = org.Mm.eg.db, keyType=&quot;ENTREZID&quot;)
  anno &lt;- as.data.frame(anno@result)
  write_csv(anno, file=paste0(&quot;output/RRBS/3_Differential_Methylation/kegg_anno_CT_&quot;, gs, &quot;.csv&quot;))

  kegg_anno_CT[[gs]] &lt;- anno
  rm(anno)
  }</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/kegg_annotation_CT-1.png" width="960" /><img src="RRBS_3_Differential_Methylation_files/figure-html/kegg_annotation_CT-2.png" width="960" /></p>
<pre class="r"><code>rm(sig.gs, sig.gs.ez)</code></pre>
</div>
</div>
<div id="cell-specificity" class="section level2 tabset tabset-fade tabset-pills">
<h2>Cell-specificity</h2>
<p>Cell type markers in the brain are taken from McKenzie et al. (Supplemental File 1, <a href="DOI:10.1038/s41598-018-27293-5" class="uri">DOI:10.1038/s41598-018-27293-5</a>). Cell specificity for a given gene is the minimum log fold change when pairwise comparing the expression of the gene in the given cell type with the expression of the gene in all other cell types. Here I’m using cell specificity score to annotate the differentially-methylated genes, with chi-square to test for biased representations of any cell type genes in either the hypo- or hyper-methylated genes, and a plot of deviations from expectation for visualization.</p>
<pre class="r"><code>cell.types &lt;- c(&quot;Astroctyes&quot;, &quot;Endothelial&quot;, &quot;Microglia&quot;, &quot;Neurons&quot;, &quot;Oligodendrocytes&quot;)
sets &lt;- c(&quot;hypo&quot;,&quot;hyper&quot;)

# read cell-specificity reference
hum.specific &lt;- read.csv(&quot;data/McKenzie_2017_Brain_Cell_Specific_Markers_Human_Specificity.csv&quot;, header=T, stringsAsFactors=F)
hum.specific &lt;- split(hum.specific, hum.specific$Celltype, drop=TRUE) 
names(hum.specific) &lt;- cell.types

# convert HGNC.ID in the specificity reference to MGI names
human = useMart(&quot;ensembl&quot;, dataset = &quot;hsapiens_gene_ensembl&quot;)
mouse = useMart(&quot;ensembl&quot;, dataset = &quot;mmusculus_gene_ensembl&quot;)

hum.specific.mg &lt;- list()
for (ct in names(hum.specific)) {
  hum.specific.mg[[ct]] = getLDS(attributes = c(&quot;hgnc_symbol&quot;), filters = &quot;hgnc_symbol&quot;, 
                              values = hum.specific[[ct]]$gene, mart = human,
                              attributesL = c(&quot;mgi_symbol&quot;), martL = mouse, uniqueRows=T)$MGI.symbol
  }</code></pre>
<div id="tt-vs-ct" class="section level3">
<h3>TT vs CT</h3>
<p>There is in general little overlap between the cell markers and our differentially-methylated genes, and no significant association between them. Hypomethylated genes are somewhat enriched for astrocytes and endothelials (BBB?) marker, and possibly neurons. The hypermethylated genes are more microglia and oligodendrocytes, but none of it is significant. Note, the chi-square test compares hypomethylated sites relative to hypermethylated sites, not within each set.</p>
<pre class="r"><code>sig.gs &lt;- snpDMgenes_TT

# get gene sets
gene.sets &lt;- lapply(sig.gs, function(l){dplyr::select(l, mgi_name)})

gene.sets.hg &lt;- list()
for (s in names(sig.gs)) {
    gene.sets.hg[[s]] = getLDS(attributes = c(&quot;mgi_symbol&quot;), filters = &quot;mgi_symbol&quot;, 
                              values = sig.gs[[s]]$mgi_name, mart = mouse,
                              attributesL = c(&quot;hgnc_symbol&quot;), martL = human, uniqueRows=T)$HGNC.symbol

}

# Number of marker genes for each cell type in each gene set
specific.set &lt;- sapply(sets, function(s) {
  sapply(cell.types, function(ct) length(intersect(hum.specific[[ct]]$gene, gene.sets.hg[[s]])), USE.NAMES=T)
})

print(specific.set)</code></pre>
<pre><code>##                  hypo hyper
## Astroctyes         95   130
## Endothelial        82    87
## Microglia          66    81
## Neurons           124   127
## Oligodendrocytes   79    92</code></pre>
<pre class="r"><code>chisq.test(specific.set)</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  specific.set
## X-squared = 2.9284, df = 4, p-value = 0.5699</code></pre>
<pre class="r"><code>assoc(t(specific.set), shade=TRUE,labeling_args = list(abbreviate_labs = c(6, 9)))</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/cell_spec_TT-1.png" width="672" /></p>
<pre class="r"><code>rm(sig.gs, gene.sets, gene.sets.hg, specific.set)</code></pre>
</div>
<div id="ct-vs-cc-3" class="section level3">
<h3>CT vs CC</h3>
<pre class="r"><code>sig.gs &lt;- snpDMgenes_CT

# get gene sets
gene.sets &lt;- lapply(sig.gs, function(l){dplyr::select(l, mgi_name)})

gene.sets.hg &lt;- list()
for (s in names(sig.gs)) {
    gene.sets.hg[[s]] = getLDS(attributes = c(&quot;mgi_symbol&quot;), filters = &quot;mgi_symbol&quot;, 
                              values = sig.gs[[s]]$mgi_name, mart = mouse,
                              attributesL = c(&quot;hgnc_symbol&quot;), martL = human, uniqueRows=T)$HGNC.symbol

}

# Number of marker genes for each cell type in each gene set
specific.set &lt;- sapply(sets, function(s) {
  sapply(cell.types, function(ct) length(intersect(hum.specific[[ct]]$gene, gene.sets.hg[[s]])), USE.NAMES=T)
})

print(specific.set)</code></pre>
<pre><code>##                  hypo hyper
## Astroctyes        106   116
## Endothelial        96    86
## Microglia          64    88
## Neurons           120   133
## Oligodendrocytes   81   105</code></pre>
<pre class="r"><code>chisq.test(specific.set)</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  specific.set
## X-squared = 4.8326, df = 4, p-value = 0.3049</code></pre>
<pre class="r"><code>assoc(t(specific.set), shade=TRUE,labeling_args = list(abbreviate_labs = c(6, 9)))</code></pre>
<p><img src="RRBS_3_Differential_Methylation_files/figure-html/cell_spec_CT-1.png" width="672" /></p>
<pre class="r"><code>rm(sig.gs, gene.sets, gene.sets.hg, specific.set)</code></pre>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>Saving differential methylation objects and kegg annotations as RDS for subsequent analyses. Saving lists of significantly methylated sites and their genes as csv.</p>
<pre class="r"><code>saveRDS(diffM_TT, file=&quot;output/RRBS/3_Differential_Methylation/diffM_TT.RDS&quot;)
saveRDS(diffM_CT, file=&quot;output/RRBS/3_Differential_Methylation/diffM_CT.RDS&quot;)

write_csv(snpDMgenes_TT[[1]], file = &quot;output/RRBS/3_Differential_Methylation/snpDMgenes_TT.csv&quot;)
write_csv(snpDMgenes_TT[[2]], file = &quot;output/RRBS/3_Differential_Methylation/snpDMgenes_TT.csv&quot;, append=TRUE)
write_csv(snpDMgenes_CT[[1]], file = &quot;output/RRBS/3_Differential_Methylation/snpDMgenes_CT.csv&quot;)
write_csv(snpDMgenes_CT[[2]], file = &quot;output/RRBS/3_Differential_Methylation/snpDMgenes_CT.csv&quot;, append=TRUE)

saveRDS(kegg_anno_TT, file=&quot;output/RRBS/3_Differential_Methylation/kegg_anno_TT.RDS&quot;)
saveRDS(kegg_anno_CT, file=&quot;output/RRBS/3_Differential_Methylation/kegg_anno_CT.RDS&quot;)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
