<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Annat Haber" />

<meta name="date" content="2021-04-03" />

<title>MTHFR RRBS Exploratory Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MTHFR C667T epigenetics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS Pilot
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_Pilot_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_Pilot_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_Pilot_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_Pilot_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RRBS
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RRBS_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RRBS_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li>
      <a href="RRBS_2_Exploratory_Analysis.html">Exploratory Analysis</a>
    </li>
    <li>
      <a href="RRBS_3_Differential_Methylation.html">Differential Methylation</a>
    </li>
    <li class="dropdown-header">Methylation by Position</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNAseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="RNAseq_0_Sample_Metadata.html">Sample Metadata</a>
    </li>
    <li>
      <a href="RNAseq_1_Processing_QC.html">Processing and QC</a>
    </li>
    <li class="dropdown-header">Exploratory Analysis</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/MTHFR_C667T">
    <span class="fab fa-github"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;annat.haber@jax.org&gt;">
    <span class="fas fa-paper-plane"></span>
     
  </a>
</li>
<li>
  <a href="https://www.dropbox.com/home/MTHFR">
    <span class="fab fa-dropbox"></span>
     
  </a>
</li>
<li>
  <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">
    <span class="fas fa-project-diagram"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">MTHFR RRBS Exploratory Analysis</h1>
<h4 class="author">Annat Haber</h4>
<h4 class="date">2021-04-03</h4>

</div>


<pre class="r"><code>rm(list=ls())

suppressPackageStartupMessages({
  library(synapser)
  #library(methylKit)
  library(tidyverse)
  library(betareg)
  library(emmeans)
  library(ComplexHeatmap)
})</code></pre>
<div id="input" class="section level2">
<h2>Input</h2>
<p>Input is percent methylation in each site for each sample (samples x sites matrix), including only sites within CpG context that are covered in all samples with at least 10x coverage.</p>
<pre class="r"><code>site_meth &lt;- readRDS(file = &quot;output/RRBS/1_Processing_QC/persite_meth.RDS&quot;)
# meta
meta &lt;- readRDS(file = &quot;output/RRBS/sample_meta.RDS&quot;)

hmc &lt;- read_tsv(&quot;data/homocysteine.txt&quot;, col_types = cols(&quot;c&quot;,&quot;n&quot;))

meta &lt;- left_join(meta, hmc, by=&quot;tag_id&quot;)</code></pre>
</div>
<div id="pattern-of-methylation" class="section level2 tabset tabset-fade tabset-pills">
<h2>Pattern of Methylation</h2>
<div id="with-outliers" class="section level3">
<h3>With Outliers</h3>
<div id="pca" class="section level4">
<h4>PCA</h4>
<pre class="r"><code>pc &lt;- prcomp(site_meth, center = TRUE)
plot(pc)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/pattern_withOutliers-1.png" width="672" /></p>
<pre class="r"><code>df &lt;- left_join(dplyr::select(meta, age, sex, genotype, sample_id),
                as_tibble(pc$x[,1:2], rownames = &quot;sample_id&quot;),
                by=&quot;sample_id&quot;)

outliers &lt;- df %&gt;% filter(abs(PC1) &gt; 2000 | abs(PC2) &gt; 2000)
  
ggplot(df, aes(x=PC1,y=PC2)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(3,1.5)) +
  geom_text(data=outliers, aes(label=sample_id), vjust = &quot;inward&quot;,  hjust = &quot;inward&quot;)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/pattern_withOutliers-2.png" width="672" /></p>
<pre class="r"><code>rm(df)</code></pre>
</div>
</div>
<div id="without-outliers" class="section level3">
<h3>Without Outliers</h3>
<pre class="r"><code>site_meth.wou &lt;- site_meth[-which(rownames(site_meth) %in% outliers$sample_id),]
meta.wou &lt;- filter(meta, !sample_id %in% outliers$sample_id)</code></pre>
<div id="pca-1" class="section level4">
<h4>PCA</h4>
<pre class="r"><code>pc &lt;- prcomp(site_meth.wou, center = TRUE)
plot(pc)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/pattern__noOutliers-1.png" width="672" /></p>
<pre class="r"><code>left_join(select(meta.wou, age, sex, genotype, sample_id),
                as_tibble(pc$x[,1:2], rownames = &quot;sample_id&quot;),
                by=&quot;sample_id&quot;) %&gt;%
  ggplot(aes(x=PC1,y=PC2)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(3,1.5)) </code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/pattern__noOutliers-2.png" width="672" /></p>
<pre class="r"><code>#+
#  geom_text(data=outliers, aes(label=sample_id), vjust = -0.5,  hjust = -0.5)</code></pre>
</div>
</div>
</div>
<div id="global-methylation" class="section level2 tabset tabset-fade tabset-pills">
<h2>Global methylation</h2>
<div id="counts" class="section level3">
<h3>Counts</h3>
<p>Calculating global methylation as the fraction of sites determined as methylated by bismark for each sample.</p>
<pre class="r"><code>global_meth_counts &lt;- read_tsv(&quot;output/RRBS/1_Processing_QC/nf_methylseq_output/bismark_summary/bismark_summary_report.txt&quot;) %&gt;%
  tibble(.name_repair = &quot;universal&quot;) %&gt;%
  mutate(sample_id = substr(File, 1, 5),
         global_meth_counts = Methylated.CpGs / (Methylated.CpGs + Unmethylated.CpGs) ) %&gt;%
  select(global_meth_counts, sample_id)

meta.gm &lt;-  left_join(meta, global_meth_counts, by=&quot;sample_id&quot;)</code></pre>
<p>Using Beta regression to test differences in global methylation as a function of sex, age, and genotype. None of the factors and interactions is significant.</p>
<pre class="r"><code>br.df &lt;- meta.gm %&gt;%
  mutate(genotype = recode(genotype, &quot;CC&quot;=0, &quot;CT&quot;=1, &quot;TT&quot;=2),
         sex = recode(sex, &quot;F&quot;=0, &quot;M&quot;=1),
         age = recode(age, &quot;6&quot;=0, &quot;18&quot;=1))
   
model = betareg(global_meth_counts ~ genotype + age + sex, data = br.df)
emmeans::joint_tests(model)</code></pre>
<pre><code>##  model term       df1 df2 F.ratio p.value
##  genotype           1 Inf   0.235 0.6278 
##  age                1 Inf   0.550 0.4582 
##  sex                1 Inf   0.315 0.5744 
##  genotype:age       1 Inf   0.162 0.6873 
##  genotype:sex       1 Inf   0.132 0.7160 
##  age:sex            1 Inf   0.202 0.6531 
##  genotype:age:sex   1 Inf   0.107 0.7442</code></pre>
<pre class="r"><code>ggplot(meta.gm, aes(x=genotype, y=global_meth_counts, color=age)) +
  geom_point() +
  facet_wrap(~ sex)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/global_meth_counts_betareg-1.png" width="672" /></p>
</div>
<div id="mean-percent" class="section level3">
<h3>Mean Percent</h3>
<p>Calculating global methylation as the mean percent methylation for each sample, averaged across sites.</p>
<pre class="r"><code>global_meth_perc &lt;- data.frame(global_meth_perc=rowMeans(site_meth)/100, sample_id=rownames(site_meth))</code></pre>
<p>Using Beta regression to test differences in global methylation as a function of sex, age, and genotype. None of the factors and interactions is significant.</p>
<pre class="r"><code>meta.gm &lt;-  left_join(meta.gm, global_meth_perc, by=&quot;sample_id&quot;)

br.df &lt;- meta.gm %&gt;%
  mutate(genotype = recode(genotype, &quot;CC&quot;=0, &quot;CT&quot;=1, &quot;TT&quot;=2),
         sex = recode(sex, &quot;F&quot;=0, &quot;M&quot;=1),
         age = recode(age, &quot;6&quot;=0, &quot;18&quot;=1))
   
model = betareg(global_meth_perc ~ genotype + age + sex, data = br.df)
emmeans::joint_tests(model)</code></pre>
<pre><code>##  model term       df1 df2 F.ratio p.value
##  genotype           1 Inf   0.221 0.6379 
##  age                1 Inf   1.159 0.2818 
##  sex                1 Inf   0.027 0.8694 
##  genotype:age       1 Inf   0.188 0.6645 
##  genotype:sex       1 Inf   0.024 0.8761 
##  age:sex            1 Inf   0.027 0.8707 
##  genotype:age:sex   1 Inf   0.024 0.8770</code></pre>
<pre class="r"><code>ggplot(meta.gm, aes(x=genotype, y=global_meth_perc, color=age)) +
  geom_point() +
  facet_wrap(~ sex)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/global_meth_percent_betareg-1.png" width="672" /></p>
</div>
<div id="correlation-with-homocysteine-level" class="section level3">
<h3>Correlation with homocysteine level</h3>
<pre class="r"><code>ggplot(meta.gm, aes(x = hmc, y = global_meth_counts)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(3,1.5)) +
  ggtitle(&quot;Counts&quot;)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/hmc-1.png" width="50%" /></p>
<pre class="r"><code>ggplot(meta.gm, aes(x = hmc, y = global_meth_perc)) + 
  geom_point(aes(shape=sex, color=genotype, size=age)) +
  scale_size_discrete(range=c(3,1.5)) +
  ggtitle(&quot;Percent&quot;)</code></pre>
<p><img src="RRBS_2_Exploratory_Analysis_files/figure-html/hmc-2.png" width="50%" /></p>
</div>
</div>
<div id="output" class="section level2">
<h2>Output</h2>
<p>Uploading sample metadata with their global methylation values to <a href="https://www.synapse.org/#!Synapse:syn23573590/wiki/607402">Synapse</a></p>
<pre class="r"><code>write_csv(meta.gm, file=&quot;output/RRBS/2_Exploratory_Analysis/RRBS_meta_globalmeth.csv&quot;)

# upload meta_global_meth to synapse
synLogin() 

file &lt;- File(&quot;output/RRBS/2_Exploratory_Analysis/RRBS_meta_globalmeth.csv&quot;,
             contentType = &quot;text/plain&quot;,
             description = &quot;RRBS sample metadata with their global methylation values&quot;,
             parent = &quot;syn25174486&quot;)

provenance &lt;- Activity(used = c(&quot;syn25174542&quot;, &quot;syn25174517&quot;),
                       executed = &quot;https://github.com/TheJacksonLaboratory/MTHFR_C667T/blob/master/RRBS_2_Exploratory_Analysis.Rmd&quot;)

file &lt;- synStore(file, activity=provenance)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
